  <div class="row flex-wrap-reverse flex-lg-wrap ms-lg-5"><div class="col-12 col-lg-9"><div class="rendered-markdown pipeline-page-content"><h1 id="introduction" class="">Introduction<a href="#introduction" class="header-link scroll_to_link"><span class="fas fa-link"></span></a></h1>
<p><strong>nf-core/hicar</strong> is a bioinformatics best-practice analysis pipeline for <a href="https://doi.org/10.1016/j.molcel.2022.01.023">HiC on Accessible Regulatory DNA (HiCAR)</a> data, a robust and sensitive assay for simultaneous measurement of chromatin accessibility and cis-regulatory chromatin contacts. Unlike the immunoprecipitation-based methods such as HiChIP, PlAC-seq and ChIA-PET, HiCAR does not require antibodies. HiCAR utilizes a Transposase-Accessible Chromatin assay to anchor the chromatin interactions. HiCAR is a tool to study chromatin interactions for low input samples and samples with no available antibodies.</p>
<p>The pipeline can also handle the experiment of HiChIP, ChIA-PET, and PLAC-Seq. It will ask user to input the peak file for the anchor peaks.</p>
<p>The pipeline is built using <a href="https://www.nextflow.io">Nextflow</a>, a workflow tool to run tasks across multiple compute infrastructures in a very portable manner. It uses Docker/Singularity containers making installation trivial and results highly reproducible. The <a href="https://www.nextflow.io/docs/latest/dsl2.html">Nextflow DSL2</a> implementation of this pipeline uses one container per process which makes it much easier to maintain and update software dependencies. Where possible, these processes have been submitted to and installed from <a href="https://github.com/nf-core/modules">nf-core/modules</a> in order to make them available to all nf-core pipelines, and to everyone within the Nextflow community!</p>
<p>On release, automated continuous integration tests run the pipeline on a full-sized dataset on the AWS cloud infrastructure. This ensures that the pipeline runs on AWS, has sensible resource allocation defaults set to run on real-world datasets, and permits the persistent storage of results to benchmark between pipeline releases and other analysis sources. The results obtained from the full-sized test can be viewed on the <a href="https://nf-co.re/hicar/results">nf-core website</a>.</p>
<h2 id="pipeline-summary" class="">Pipeline summary<a href="#pipeline-summary" class="header-link scroll_to_link"><span class="fas fa-link"></span></a></h2>
<ol>
<li>Read QC (<a href="https://www.bioinformatics.babraham.ac.uk/projects/fastqc/"><code>FastQC</code></a>)</li>
<li>Trim reads (<a href="https://cutadapt.readthedocs.io/en/stable/"><code>cutadapt</code></a>)</li>
<li>Map reads (<a href="http://bio-bwa.sourceforge.net/bwa.shtml"><code>bwa mem</code></a>)</li>
<li>Filter reads (<a href="https://pairtools.readthedocs.io/en/latest/"><code>pairtools</code></a>)</li>
<li>Quality analysis (<a href="https://github.com/4dn-dcic/pairsqc"><code>pairsqc</code></a>)</li>
<li>Call peaks for ATAC reads (R2 reads) (<a href="https://macs3-project.github.io/MACS/"><code>MACS2</code></a>) and/or call peaks for R1 reads.</li>
<li>Find genomic interaction loops (<a href="https://github.com/ijuric/MAPS"><code>MAPS</code></a>)</li>
<li>Differential analysis (<a href="https://bioconductor.org/packages/edgeR/"><code>edgeR</code></a>)</li>
<li>Annotate genomic interaction loops (<a href="https://bioconductor.org/packages/ChIPpeakAnno/"><code>ChIPpeakAnno</code></a>)</li>
<li>Create cooler files (<a href="https://cooler.readthedocs.io/en/latest/index.html"><code>cooler</code></a>, .hic files <a href="https://github.com/aidenlab/juicer/wiki"><code>Juicer_tools</code></a>, and circos files <a href="http://circos.ca/"><code>circos</code></a>) for visualization.</li>
<li>Present QC for raw reads (<a href="http://multiqc.info/"><code>MultiQC</code></a>)</li>
</ol>
<p><img src="https://raw.githubusercontent.com/jianhong/hicar/1.0.0/./docs/images/workflow.svg" alt="work flow of the pipeline" /></p>
<h2 id="quick-start" class="">Quick Start<a href="#quick-start" class="header-link scroll_to_link"><span class="fas fa-link"></span></a></h2>
<ol>
<li>
<p>Install <a href="https://www.nextflow.io/docs/latest/getstarted.html#installation"><code>Nextflow</code></a> (<code>&gt;=21.10.3</code>)</p>
</li>
<li>
<p>Install any of <a href="https://docs.docker.com/engine/installation/"><code>Docker</code></a>, <a href="https://www.sylabs.io/guides/3.0/user-guide/"><code>Singularity</code></a> (you can follow <a href="https://singularity-tutorial.github.io/01-installation/">this tutorial</a>), <a href="https://podman.io/"><code>Podman</code></a>, <a href="https://nersc.gitlab.io/development/shifter/how-to-use/"><code>Shifter</code></a> or <a href="https://hpc.github.io/charliecloud/"><code>Charliecloud</code></a> for full pipeline reproducibility <em>(you can use <a href="https://conda.io/miniconda.html"><code>Conda</code></a> both to install Nextflow itself and also to manage software within pipelines. Please only use it within pipelines as a last resort; see <a href="https://nf-co.re/usage/configuration#basic-configuration-profiles">docs</a>)</em>.</p>
</li>
<li>
<p>Download the pipeline and test it on a minimal dataset with a single command:</p>
<pre><code class="language-console">nextflow run nf-core/hicar -profile test,YOURPROFILE --outdir &lt;OUTDIR&gt;</code></pre>
<p>Note that some form of configuration will be needed so that Nextflow knows how to fetch the required software. This is usually done in the form of a config profile (<code>YOURPROFILE</code> in the example command above). You can chain multiple config profiles in a comma-separated string.</p>
<blockquote>
<ul>
<li>The pipeline comes with config profiles called <code>docker</code>, <code>singularity</code>, <code>podman</code>, <code>shifter</code>, <code>charliecloud</code> and <code>conda</code> which instruct the pipeline to use the named tool for software management. For example, <code>-profile test,docker</code>.</li>
<li>Please check <a href="https://github.com/nf-core/configs#documentation">nf-core/configs</a> to see if a custom config file to run nf-core pipelines already exists for your Institute. If so, you can simply use <code>-profile &lt;institute&gt;</code> in your command. This will enable either <code>docker</code> or <code>singularity</code> and set the appropriate execution settings for your local compute environment.</li>
<li>If you are using <code>singularity</code>, please use the <a href="https://nf-co.re/tools/#downloading-pipelines-for-offline-use"><code>nf-core download</code></a> command to download images first, before running the pipeline. Setting the <a href="https://www.nextflow.io/docs/latest/singularity.html?#singularity-docker-hub"><code>NXF_SINGULARITY_CACHEDIR</code> or <code>singularity.cacheDir</code></a> Nextflow options enables you to store and re-use the images from a central location for future pipeline runs.</li>
<li>If you are using <code>conda</code>, it is highly recommended to use the <a href="https://www.nextflow.io/docs/latest/conda.html"><code>NXF_CONDA_CACHEDIR</code> or <code>conda.cacheDir</code></a> settings to store the environments in a central location for future pipeline runs.</li>
</ul>
</blockquote>
</li>
<li>
<p>Start running your own analysis!</p>
<pre><code class="language-console">nextflow run nf-core/hicar -profile &lt;docker/singularity/podman/shifter/charliecloud/conda/institute&gt; \
    --input samples.csv \   # Input data
    --qval_thresh 0.01 \    # Cut-off q-value for MACS2
    --genome GRCh38 \       # Genome Reference
    --mappability /path/mappability/bigWig/file  # Provide mappability to avoid memory intensive calculation</code></pre>
<p>Run it on cluster.</p>
<p>First prepare a profile config file named as <a href="https://nf-co.re/hicar/usage">profile.config</a> and a <a href="https://nf-co.re/hicar/usage">samplesheet</a>.
Then run:</p>
<pre><code class="language-console">nextflow run nf-core/hicar --input samplesheet.csv --outdir &lt;OUTDIR&gt; --genome GRCh37 -profile &lt;docker/singularity/podman/shifter/charliecloud/conda/institute&gt;</code></pre>
</li>
</ol>
<h2 id="documentation" class="">Documentation<a href="#documentation" class="header-link scroll_to_link"><span class="fas fa-link"></span></a></h2>
<p>The nf-core/hicar pipeline comes with documentation about the pipeline <a href="https://nf-co.re/hicar/usage">usage</a>, <a href="https://nf-co.re/hicar/parameters">parameters</a> and <a href="https://nf-co.re/hicar/output">output</a>.</p>
<h2 id="credits" class="">Credits<a href="#credits" class="header-link scroll_to_link"><span class="fas fa-link"></span></a></h2>
<p>nf-core/hicar was originally written by Jianhong Ou, <a href="https://github.com/yuxuth">Yu Xiang</a>, and Yarui Diao.</p>
<p>We thank the following people for their extensive assistance in the development of this pipeline: Phil Ewels, Mahesh Binzer-Panchal and Friederike Hanssen.</p>
<h2 id="contributions-and-support" class="">Contributions and Support<a href="#contributions-and-support" class="header-link scroll_to_link"><span class="fas fa-link"></span></a></h2>
<p>If you would like to contribute to this pipeline, please see the <a href="/hicar/1.0.0/./.github/CONTRIBUTING">contributing guidelines</a>.</p>
<p>For further information or help, don't hesitate to get in touch on the <a href="https://nfcore.slack.com/channels/hicar">Slack <code>#hicar</code> channel</a> (you can join with <a href="https://nf-co.re/join/slack">this invite</a>).</p>
<h2 id="citations" class="">Citations<a href="#citations" class="header-link scroll_to_link"><span class="fas fa-link"></span></a></h2>
<p>If you use nf-core/hicar for your analysis, please cite it using the following doi: <a href="https://doi.org/10.5281/zenodo.6515312">10.5281/zenodo.6515312</a></p>
<p>An extensive list of references for the tools used by the pipeline can be found in the <a href="/hicar/1.0.0/./CITATIONS"><code>CITATIONS.md</code></a> file.</p>
<p>You can cite the <code>nf-core</code> publication as follows:</p>
<blockquote>
<p><strong>The nf-core framework for community-curated bioinformatics pipelines.</strong></p>
<p>Philip Ewels, Alexander Peltzer, Sven Fillinger, Harshil Patel, Johannes Alneberg, Andreas Wilm, Maxime Ulysse Garcia, Paolo Di Tommaso &amp; Sven Nahnsen.</p>
<p><em>Nat Biotechnol.</em> 2020 Feb 13. doi: <a href="https://dx.doi.org/10.1038/s41587-020-0439-x">10.1038/s41587-020-0439-x</a>.</p>
</blockquote></div></div><div class="col-12 col-lg-3 ps-2"><div class="side-sub-subnav sticky-top">
<div class="pipeline-sidebar">
  <div class="row border-bottom pb-2">
    <div class="col-12">
      <h6><i class="fas fa-terminal fa-xs"></i> command</h6>
      <div class="input-group input-group-sm pipeline-run-cmd">
        <input type="text" class="form-control input-sm code" id="pipeline-run-cmd-text" data-autoselect="" value="nextflow run jianhong/hicar -r 1.0.0 -profile test" aria-label="Copy run command" readonly="">
        <button class="btn btn-outline-secondary copy-txt" data-bs-target="pipeline-run-cmd-text" data-bs-toggle="tooltip" data-bs-placement="left" title="Copy to clipboard" type="button"><i class="fas fa-clipboard px-1"></i></button>
      </div>
    </div>
  </div>
  <div>
    <h6>get in touch</h6>
    <p><a class="btn btn-sm btn-outline-info" href="https://nfcore.slack.com/channels/hicar"><i class="fab fa-slack me-1"></i> Ask a question on Slack</a></p>
    <p><a class="btn btn-sm btn-outline-secondary" href="https://github.com/jianhong/hicar/issues"><i class="fab fa-github me-1"></i> Open an issue on GitHub</a></p>
  </div>
<div>
    <h6>quick start with test run</h6>
    <p><a href="https://youtu.be/XpB8tRhPrqk"><img role="img" aria-label="Youtube Video" src="https://img.youtube.com/vi/XpB8tRhPrqk/0.jpg" alt="Youtube Video"></a></p>
</div>
</div>

</div></div></div>