  <div class="row flex-wrap-reverse flex-lg-wrap ms-lg-5"><div class="col-12 col-lg-9"><div class="rendered-markdown pipeline-page-content"><h1 id="introduction" class="">Introduction<a href="#introduction" class="header-link scroll_to_link"><span class="fas fa-link"></span></a></h1>
<p>HiC on Accessible Regulatory DNA (HiCAR) is a feasible tool to study the long-range interactions
of the cis- and trans-regulatory elements (cREs/TREs) especially for the low input
samples and the samples with no available antibodies. To facilitate the analysis of HiCAR
data, the Nextflow platform based nf-core/hicar pipeline can detect the chromatin interaction
in single nucleotide resolution within and among chromosomes.</p>
<h2 id="samplesheet-input" class="">Samplesheet input<a href="#samplesheet-input" class="header-link scroll_to_link"><span class="fas fa-link"></span></a></h2>
<p>You will need to create a samplesheet with information about the samples you would like to analzse before running the pipeline. Use this parameter to specify its location. It has to be a comma-separated file with 4 columns, and a header row as shown in the examples below.</p>
<pre><code class="language-console"><span class="text-nowrap">--input</span> '[path to samplesheet file]'</code></pre>
<h3 id="multiple-runs-of-the-same-sample" class="">Multiple runs of the same sample<a href="#multiple-runs-of-the-same-sample" class="header-link scroll_to_link"><span class="fas fa-link"></span></a></h3>
<p>The <code>sample</code> identifiers have to be the same when you have re-sequenced the same sample more than once e.g. to increase sequencing depth. The pipeline will concatenate the raw reads before performing any downstream analysis. Below is an example for the same sample sequenced across 3 lanes:</p>
<pre><code class="language-console">group,replicate,fastq_1,fastq_2
CONTROL,1,AEG588A1_S1_L002_R1_001.fastq.gz,AEG588A1_S1_L002_R2_001.fastq.gz
CONTROL,1,AEG588A1_S1_L003_R1_001.fastq.gz,AEG588A1_S1_L003_R2_001.fastq.gz
CONTROL,1,AEG588A1_S1_L004_R1_001.fastq.gz,AEG588A1_S1_L004_R2_001.fastq.gz</code></pre>
<h3 id="full-samplesheet" class="">Full samplesheet<a href="#full-samplesheet" class="header-link scroll_to_link"><span class="fas fa-link"></span></a></h3>
<p>HiCAR requires paired-end sequencing. Both fastq_1 and fastq_2 must be provided in the samplesheet. The samplesheet can have as many columns as you desire, however, there is a strict requirement for the first 4 columns to match those defined in the table below.</p>
<p>If md5_1/2 is provided, the pipeline will check the checksums.</p>
<p>A final samplesheet file consisting of two groups of paired-end data may look something like the one below. This is for 6 samples, where <code>TREATMENT_REP3</code> has been sequenced twice.</p>
<pre><code class="language-console">group,replicate,fastq_1,fastq_2,md5_1,md5_2
CONTROL,1,AEG588A1_S1_L002_R1_001.fastq.gz,AEG588A1_S1_L002_R2_001.fastq.gz,,
CONTROL,2,AEG588A2_S2_L002_R1_001.fastq.gz,AEG588A2_S2_L002_R2_001.fastq.gz,,
CONTROL,3,AEG588A3_S3_L002_R1_001.fastq.gz,AEG588A3_S3_L002_R2_001.fastq.gz,,
TREATMENT,1,AEG588A4_S4_L003_R1_001.fastq.gz,AEG588A4_S4_L003_R2_001.fastq.gz,,
TREATMENT,2,AEG588A5_S5_L003_R1_001.fastq.gz,AEG588A5_S5_L003_R2_001.fastq.gz,,
TREATMENT,3,AEG588A6_S6_L003_R1_001.fastq.gz,AEG588A6_S6_L003_R2_001.fastq.gz,,
TREATMENT,3,AEG588A6_S6_L004_R1_001.fastq.gz,AEG588A6_S6_L004_R2_001.fastq.gz,,</code></pre>
<div class="table-responsive"><table class="table table-bordered table-striped table-sm small">
<thead>
<tr>
<th>Column</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr>
<td><code>group</code></td>
<td>Custom group name. This entry will be identical for multiple sequencing libraries/runs from the same sample. Spaces in sample names are automatically converted to underscores (<code>_</code>).</td>
</tr>
<tr>
<td><code>replicate</code></td>
<td>Biological replicates of the samples.</td>
</tr>
<tr>
<td><code>fastq_1</code></td>
<td>Full path to FastQ file for Illumina short reads 1. File has to be gzipped and have the extension ".fastq.gz" or ".fq.gz".</td>
</tr>
<tr>
<td><code>fastq_2</code></td>
<td>Full path to FastQ file for Illumina short reads 2. File has to be gzipped and have the extension ".fastq.gz" or ".fq.gz".</td>
</tr>
<tr>
<td><code>md5_1</code></td>
<td>Checksum for fastq_1. The checksums of the files will be check to make sure the file is not truncated if provided.</td>
</tr>
<tr>
<td><code>md5_2</code></td>
<td>Checksum for fastq_2. The checksums of the files will be check to make sure the file is not truncated if provided.</td>
</tr>
</tbody>
</table></div>
<p>An <a href="/hicar/1.0.0/docs/../assets/samplesheet.csv">example samplesheet</a> has been provided with the pipeline.</p>
<h2 id="running-the-pipeline" class="">Running the pipeline<a href="#running-the-pipeline" class="header-link scroll_to_link"><span class="fas fa-link"></span></a></h2>
<p>The typical command for running the pipeline is as follows:</p>
<pre><code class="language-console">nextflow run nf-core/hicar --input samplesheet.csv --outdir &lt;OUTDIR&gt; --genome GRCh37 -profile docker</code></pre>
<p>This will launch the pipeline with the <code>docker</code> configuration profile. See below for more information about profiles.</p>
<p>Note that the pipeline will create the following files in your working directory:</p>
<pre><code class="language-console">work                # Directory containing the nextflow working files
&lt;OUTIDR&gt;            # Finished results in specified location (defined with --outdir)
.nextflow_log       # Log file from Nextflow
# Other nextflow hidden files, eg. history of pipeline runs and old logs.</code></pre>
<h3 id="updating-the-pipeline" class="">Updating the pipeline<a href="#updating-the-pipeline" class="header-link scroll_to_link"><span class="fas fa-link"></span></a></h3>
<p>When you run the above command, Nextflow automatically pulls the pipeline code from GitHub and stores it as a cached version. When running the pipeline after this, it will always use the cached version if available - even if the pipeline has been updated since. To make sure that you're running the latest version of the pipeline, make sure that you regularly update the cached version of the pipeline:</p>
<pre><code class="language-console">nextflow pull nf-core/hicar</code></pre>
<h3 id="reproducibility" class="">Reproducibility<a href="#reproducibility" class="header-link scroll_to_link"><span class="fas fa-link"></span></a></h3>
<p>It is a good idea to specify a pipeline version when running the pipeline on your data. This ensures that a specific version of the pipeline code and software are used when you run your pipeline. If you keep using the same tag, you'll be running the same version of the pipeline, even if there have been changes to the code since.</p>
<p>First, go to the <a href="https://github.com/nf-core/hicar/releases">nf-core/hicar releases page</a> and find the latest version number - numeric only (eg. <code>1.0.0</code>). Then specify this when running the pipeline with <code>-r</code> (one hyphen) - eg. <code>-r 1.0.0</code>.</p>
<p>This version number will be logged in reports when you run the pipeline, so that you'll know what you used when you look back in the future.</p>
<h2 id="core-nextflow-arguments" class="">Core Nextflow arguments<a href="#core-nextflow-arguments" class="header-link scroll_to_link"><span class="fas fa-link"></span></a></h2>
<blockquote>
<p><strong>NB:</strong> These options are part of Nextflow and use a <em>single</em> hyphen (pipeline parameters use a double-hyphen).</p>
</blockquote>
<h3 id="profile" class=""><code>-profile</code><a href="#profile" class="header-link scroll_to_link"><span class="fas fa-link"></span></a></h3>
<p>Use this parameter to choose a configuration profile. Profiles can give configuration presets for different compute environments.</p>
<p>Several generic profiles are bundled with the pipeline which instruct the pipeline to use software packaged using different methods (Docker, Singularity, Podman, Shifter, Charliecloud, Conda) - see below. When using Biocontainers, most of these software packaging methods pull Docker containers from quay.io e.g <a href="https://quay.io/repository/biocontainers/fastqc">FastQC</a> except for Singularity which directly downloads Singularity images via https hosted by the <a href="https://depot.galaxyproject.org/singularity/">Galaxy project</a> and Conda which downloads and installs software locally from <a href="https://bioconda.github.io/">Bioconda</a>.</p>
<blockquote>
<p>We highly recommend the use of Docker or Singularity containers for full pipeline reproducibility, however when this is not possible, Conda is also supported.</p>
</blockquote>
<p>The pipeline also dynamically loads configurations from <a href="https://github.com/nf-core/configs">https://github.com/nf-core/configs</a> when it runs, making multiple config profiles for various institutional clusters available at run time. For more information and to see if your system is available in these configs please see the <a href="https://github.com/nf-core/configs#documentation">nf-core/configs documentation</a>.</p>
<p>Note that multiple profiles can be loaded, for example: <code>-profile test,docker</code> - the order of arguments is important!
They are loaded in sequence, so later profiles can overwrite earlier profiles.</p>
<p>If <code>-profile</code> is not specified, the pipeline will run locally and expect all software to be installed and available on the <code>PATH</code>. This is <em>not</em> recommended.</p>
<ul>
<li><code>docker</code><ul>
<li>A generic configuration profile to be used with <a href="https://docker.com/">Docker</a></li>
</ul>
</li>
<li><code>singularity</code><ul>
<li>A generic configuration profile to be used with <a href="https://sylabs.io/docs/">Singularity</a></li>
</ul>
</li>
<li><code>podman</code><ul>
<li>A generic configuration profile to be used with <a href="https://podman.io/">Podman</a></li>
</ul>
</li>
<li><code>shifter</code><ul>
<li>A generic configuration profile to be used with <a href="https://nersc.gitlab.io/development/shifter/how-to-use/">Shifter</a></li>
</ul>
</li>
<li><code>charliecloud</code><ul>
<li>A generic configuration profile to be used with <a href="https://hpc.github.io/charliecloud/">Charliecloud</a></li>
</ul>
</li>
<li><code>conda</code><ul>
<li>A generic configuration profile to be used with <a href="https://conda.io/docs/">Conda</a>. Please only use Conda as a last resort i.e. when it's not possible to run the pipeline with Docker, Singularity, Podman, Shifter or Charliecloud.</li>
</ul>
</li>
<li><code>test</code><ul>
<li>A profile with a complete configuration for automated testing</li>
<li>Includes links to test data so needs no other parameters</li>
</ul>
</li>
</ul>
<h3 id="resume" class=""><code>-resume</code><a href="#resume" class="header-link scroll_to_link"><span class="fas fa-link"></span></a></h3>
<p>Specify this when restarting a pipeline. Nextflow will use cached results from any pipeline steps where the inputs are the same, continuing from where it got to previously. For input to be considered the same, not only the names must be identical but the files' contents as well. For more info about this parameter, see <a href="https://www.nextflow.io/blog/2019/demystifying-nextflow-resume.html">this blog post</a>.</p>
<p>You can also supply a run name to resume a specific run: <code>-resume [run-name]</code>. Use the <code>nextflow log</code> command to show previous run names.</p>
<h3 id="c" class=""><code>-c</code><a href="#c" class="header-link scroll_to_link"><span class="fas fa-link"></span></a></h3>
<p>Specify the path to a specific config file (this is a core Nextflow command). See the <a href="https://nf-co.re/usage/configuration">nf-core website documentation</a> for more information.</p>
<h2 id="custom-configuration" class="">Custom configuration<a href="#custom-configuration" class="header-link scroll_to_link"><span class="fas fa-link"></span></a></h2>
<h3 id="resource-requests" class="">Resource requests<a href="#resource-requests" class="header-link scroll_to_link"><span class="fas fa-link"></span></a></h3>
<p>Whilst the default requirements set within the pipeline will hopefully work for most people and with most input data, you may find that you want to customise the compute resources that the pipeline requests. Each step in the pipeline has a default set of requirements for number of CPUs, memory and time. For most of the steps in the pipeline, if the job exits with any of the error codes specified <a href="https://github.com/nf-core/rnaseq/blob/4c27ef5610c87db00c3c5a3eed10b1d161abf575/conf/base.config#L18">here</a> it will automatically be resubmitted with higher requests (2 x original, then 3 x original). If it still fails after the third attempt then the pipeline execution is stopped.</p>
<p>For example, if the nf-core/hicar pipeline is failing after multiple re-submissions of the <code>MAPS_CUT</code> process due to an exit code of <code>137</code> this would indicate that there is an out of memory issue:</p>
<pre><code class="language-console">[9d/172ca5] NOTE: Process `NFCORE_HICAR:HICAR:MAPS_MULTIENZYME:MAPS_CUT` terminated with an error exit status (137) -- Execution is retried (1)
Error executing process &gt; 'NFCORE_HICAR:HICAR:MAPS_MULTIENZYME:MAPS_CUT (10000)'

Caused by:
    Process `NFCORE_HICAR:HICAR:MAPS_MULTIENZYME:MAPS_CUT (10000)` terminated with an error exit status (137)

Command executed:
    restriction_cut_multipleenzyme.py \
        -f chr22.fa \
        -s GTAC \
        -p 1 \
        -b 10000 \
        -o 10000_CviQI.cut \
        -c 2

Command exit status:
    137

Command output:
    (empty)</code></pre>
<p>To bypass this error you would need to find exactly which resources are set by the <code>MAPS_CUT</code> process. The quickest way is to search for <code>process MAPS_CUT</code> in the <a href="https://github.com/nf-core/hicar/search?q=process+MAPS_CUT">nf-core/hicar Github repo</a>. We have standardised the structure of Nextflow DSL2 pipelines such that all module files will be present in the <code>modules/</code> directory and so based on the search results the file we want is <code>modules/local/maps/cut.nf</code>. If you click on the link to that file you will notice that there is a <code>label</code> directive at the top of the module that is set to <a href="https://github.com/nf-core/hicar/blob/master/modules/local/maps/cut.nf#L8"><code>label process_high</code></a>. The <a href="https://www.nextflow.io/docs/latest/process.html#label">Nextflow <code>label</code></a> directive allows us to organise workflow processes in separate groups which can be referenced in a configuration file to select and configure subset of processes having similar computing requirements. The default values for the <code>process_high</code> label are set in the pipeline's <a href="https://github.com/nf-core/hicar/blob/master/conf/base.config#L36-L40"><code>base.config</code></a> which in this case is defined as 72GB. Providing you haven't set any other standard nf-core parameters to <strong>cap</strong> the <a href="https://nf-co.re/usage/configuration#max-resources">maximum resources</a> used by the pipeline then we can try and bypass the <code>MAPS_CUT</code> process failure by creating a custom config file that sets at least 72GB of memory, in this case increased to 100GB. The custom config below can then be provided to the pipeline via the <a href="#-c"><code>-c</code></a> parameter as highlighted in previous sections.</p>
<pre><code class="language-groovy">process {
    withName: MAPS_CUT {
        memory = 100.GB
    }
}</code></pre>
<blockquote>
<p><strong>NB:</strong> We specify the full process name i.e. <code>NFCORE_RNASEQ:RNASEQ:ALIGN_STAR:STAR_ALIGN</code> in the config file because this takes priority over the short name (<code>STAR_ALIGN</code>) and allows existing configuration using the full process name to be correctly overridden.</p>
<p>If you get a warning suggesting that the process selector isn't recognised check that the process name has been specified correctly.</p>
</blockquote>
<h3 id="updating-containers" class="">Updating containers<a href="#updating-containers" class="header-link scroll_to_link"><span class="fas fa-link"></span></a></h3>
<p>The <a href="https://www.nextflow.io/docs/latest/dsl2.html">Nextflow DSL2</a> implementation of this pipeline uses one container per process which makes it much easier to maintain and update software dependencies. If for some reason you need to use a different version of a particular tool with the pipeline then you just need to identify the <code>process</code> name and override the Nextflow <code>container</code> definition for that process using the <code>withName</code> declaration. For example, in the <a href="https://nf-co.re/viralrecon">nf-core/viralrecon</a> pipeline a tool called <a href="https://github.com/cov-lineages/pangolin">Pangolin</a> has been used during the COVID-19 pandemic to assign lineages to SARS-CoV-2 genome sequenced samples. Given that the lineage assignments change quite frequently it doesn't make sense to re-release the nf-core/viralrecon everytime a new version of Pangolin has been released. However, you can override the default container used by the pipeline by creating a custom config file and passing it as a command-line argument via <code>-c custom.config</code>.</p>
<ol>
<li>
<p>Check the default version used by the pipeline in the module file for <a href="https://github.com/nf-core/viralrecon/blob/a85d5969f9025409e3618d6c280ef15ce417df65/modules/nf-core/software/pangolin/main.nf#L14-L19">Pangolin</a></p>
</li>
<li>
<p>Find the latest version of the Biocontainer available on <a href="https://quay.io/repository/biocontainers/pangolin?tag=latest&amp;tab=tags">Quay.io</a></p>
</li>
<li>
<p>Create the custom config accordingly:</p>
<ul>
<li>
<p>For Docker:</p>
<pre><code class="language-groovy">process {
    withName: PANGOLIN {
        container = 'quay.io/biocontainers/pangolin:3.0.5--pyhdfd78af_0'
    }
}</code></pre>
</li>
<li>
<p>For Singularity:</p>
<pre><code class="language-groovy">process {
    withName: PANGOLIN {
        container = 'https://depot.galaxyproject.org/singularity/pangolin:3.0.5--pyhdfd78af_0'
    }
}</code></pre>
</li>
<li>
<p>For Conda:</p>
<pre><code class="language-groovy">process {
    withName: PANGOLIN {
        conda = 'bioconda::pangolin=3.0.5'
    }
}</code></pre>
</li>
</ul>
</li>
</ol>
<blockquote>
<p><strong>NB:</strong> If you wish to periodically update individual tool-specific results (e.g. Pangolin) generated by the pipeline then you must ensure to keep the <code>work/</code> directory otherwise the <code>-resume</code> ability of the pipeline will be compromised and it will restart from scratch.</p>
</blockquote>
<h3 id="nf-coreconfigs" class="">nf-core/configs<a href="#nf-coreconfigs" class="header-link scroll_to_link"><span class="fas fa-link"></span></a></h3>
<p>In most cases, you will only need to create a custom config as a one-off but if you and others within your organisation are likely to be running nf-core pipelines regularly and need to use the same settings regularly it may be a good idea to request that your custom config file is uploaded to the <code>nf-core/configs</code> git repository. Before you do this please can you test that the config file works with your pipeline of choice using the <code>-c</code> parameter. You can then create a pull request to the <code>nf-core/configs</code> repository with the addition of your config file, associated documentation file (see examples in <a href="https://github.com/nf-core/configs/tree/master/docs"><code>nf-core/configs/docs</code></a>), and amending <a href="https://github.com/nf-core/configs/blob/master/nfcore_custom.config"><code>nfcore_custom.config</code></a> to include your custom profile.</p>
<p>See the main <a href="https://www.nextflow.io/docs/latest/config.html">Nextflow documentation</a> for more information about creating your own configuration files.</p>
<p>If you have any questions or issues please send us a message on <a href="https://nf-co.re/join/slack">Slack</a> on the <a href="https://nfcore.slack.com/channels/configs"><code>#configs</code> channel</a>.</p>
<h2 id="running-in-the-background" class="">Running in the background<a href="#running-in-the-background" class="header-link scroll_to_link"><span class="fas fa-link"></span></a></h2>
<p>Nextflow handles job submissions and supervises the running jobs. The Nextflow process must run until the pipeline is finished.</p>
<p>The Nextflow <code>-bg</code> flag launches Nextflow in the background, detached from your terminal so that the workflow does not stop if you log out of your session. The logs are saved to a file.</p>
<p>Alternatively, you can use <code>screen</code> / <code>tmux</code> or similar tool to create a detached session which you can log back into at a later time.
Some HPC setups also allow you to run nextflow within a cluster job submitted your job scheduler (from where it submits more jobs).</p>
<h2 id="nextflow-memory-requirements" class="">Nextflow memory requirements<a href="#nextflow-memory-requirements" class="header-link scroll_to_link"><span class="fas fa-link"></span></a></h2>
<p>In some cases, the Nextflow Java virtual machines can start to request a large amount of memory.
We recommend adding the following line to your environment to limit this (typically in <code>~/.bashrc</code> or <code>~./bash_profile</code>):</p>
<pre><code class="language-console">NXF_OPTS='-Xms1g -Xmx4g'</code></pre>
<h2 id="troubleshooting" class="">Troubleshooting<a href="#troubleshooting" class="header-link scroll_to_link"><span class="fas fa-link"></span></a></h2>
<ul>
<li>Error: <code>The exit status of the task that caused the workflow execution to fail was: null.</code></li>
</ul>
<p>Check the files are readable for the workflow.</p>
<ul>
<li>Error: <code>Session aborted -- Cause: Unable to execute HTTP request: ngi-igenomes.s3.amazonaws.com</code></li>
</ul>
<p>The internet connection reached the limitation. Try to resume the analysis one hour later.</p>
<ul>
<li>Error: <code>PaddingError: Placeholder of length '80' too short in package</code></li>
</ul>
<p>There is no easy answer here. The new <code>conda</code> packages should having a longer prefix (255 characters).
The possible solution now is that try to run the pipeline in a shorter folder path, if at all possible.</p>
<ul>
<li>Error: <code>Not a conda environment</code> or <code>command not found</code></li>
</ul>
<p>There is something going wrong with the conda environment building.
Just try to remove the conda environment folder and resume the run.</p>
<ul>
<li>Error: <code>unable to load shared object 'work/conda/env-xxxxxx/lib/R/library/rtracklayer/libs/rtracklayer.dylib', dlopen(rtracklayer.dylib, 6) Library not loaded: @rpath/libssl.1.1.dylib</code></li>
</ul>
<p>The openssl installation have issues for <code>conda</code>. Try to reinstall it by
<code>conda activate work/conda/env-xxxxxx &amp;&amp; conda install --force-reinstall -y openssl</code></p>
<ul>
<li>Error: <code>error Can't locate Statistics/Basic.pm</code></li>
</ul>
<p>The perl-statistics-basic installed in wrong location. Try to reinstall it by
<code>conda activate work/conda/env-xxxxx &amp;&amp; perl -MCPAN -e 'CPAN::install(Statistics::Basic)'</code></p>
<ul>
<li><code>Error in result[[njob]] &lt;- value : attempt to select less than one element in OneIndex</code></li>
</ul>
<p>The error may caused by out of memory (although the error message seems to be unrelated to memory). Try to set <code><span class="text-nowrap">--peak_pair_block</code></span> to a smaller number less than 1e9.</p>
<h3 id="known-issue-with-juicer_tools" class="">Known issue with Juicer_tools<a href="#known-issue-with-juicer_tools" class="header-link scroll_to_link"><span class="fas fa-link"></span></a></h3>
<p>If you are using <a href="https://github.com/aidenlab/juicer/wiki/">Juicer_tools</a> with GPU supported, it is not supported by the containers. We are using <a href="https://github.com/aidenlab/juicer/wiki/Pre">Juicer Tools Pre</a> to create the <a href="https://doi.org/10.1016/j.cels.2016.07.002">hic files</a> from aligned HiCAR reads.
We recommend having at least 4GB free RAM to generate the hic files.</p></div></div><div class="col-12 col-lg-3 ps-2"><div class="side-sub-subnav sticky-top"><nav class="toc pt-2 auto-toc border-start d-print-none"><strong class="ms-3 d-inline-block w-100 text-secondary border-bottom">On this page</strong>
<nav class="nav flex-column flex-nowrap">
<a class="nav-link scroll_to_link py-1  active " href="#introduction">Introduction</a>
<nav class="nav flex-column flex-nowrap">
<a class="nav-link scroll_to_link py-1 " href="#samplesheet-input">Samplesheet input</a>
<nav class="nav flex-column flex-nowrap">
<a class="nav-link scroll_to_link py-1 " href="#multiple-runs-of-the-same-sample">Multiple runs of the same sample</a>
<a class="nav-link scroll_to_link py-1 " href="#full-samplesheet">Full samplesheet</a>
</nav>

<a class="nav-link scroll_to_link py-1 " href="#running-the-pipeline">Running the pipeline</a>
<nav class="nav flex-column flex-nowrap">
<a class="nav-link scroll_to_link py-1 " href="#updating-the-pipeline">Updating the pipeline</a>
<a class="nav-link scroll_to_link py-1 " href="#reproducibility">Reproducibility</a>
</nav>

<a class="nav-link scroll_to_link py-1 " href="#core-nextflow-arguments">Core Nextflow arguments</a>
<nav class="nav flex-column flex-nowrap">
<a class="nav-link scroll_to_link py-1 " href="#profile"><code>-profile</code></a>
<a class="nav-link scroll_to_link py-1 " href="#resume"><code>-resume</code></a>
<a class="nav-link scroll_to_link py-1 " href="#c"><code>-c</code></a>
</nav>

<a class="nav-link scroll_to_link py-1 " href="#custom-configuration">Custom configuration</a>
<nav class="nav flex-column flex-nowrap">
<a class="nav-link scroll_to_link py-1 " href="#resource-requests">Resource requests</a>
<a class="nav-link scroll_to_link py-1 " href="#updating-containers">Updating containers</a>
<a class="nav-link scroll_to_link py-1 " href="#nf-coreconfigs">nf-core/configs</a>
</nav>

<a class="nav-link scroll_to_link py-1 " href="#running-in-the-background">Running in the background</a>
<a class="nav-link scroll_to_link py-1 " href="#nextflow-memory-requirements">Nextflow memory requirements</a>
<a class="nav-link scroll_to_link py-1 " href="#troubleshooting">Troubleshooting</a>
<nav class="nav flex-column flex-nowrap">
<a class="nav-link scroll_to_link py-1 " href="#known-issue-with-juicer_tools">Known issue with Juicer_tools</a></nav></nav></nav><p class="small text-end mt-3"><a href="#" class="text-muted"><i class="fas fa-arrow-to-top"></i> Back to top</a></p></nav></div></div></div>