  <div class="row flex-wrap-reverse flex-lg-wrap ms-lg-5"><div class="col-12 col-lg-9"><div class="rendered-markdown pipeline-page-content"><div class="param-docs">
    <h1 id="command" class="">
      Command
      <a href="#command" class="header-link"><span class="fas fa-link"></span></a>
    </h1>
    <p>
    The top 3 questions need to be answered before you run the pipeline:
      <ol>
          <li>What is the first enzyme, eg. CviQI, and splintoligo you used in the experiment?
          The sequence will be used to determine the "--enzyme", "--cutadapt-5end" and "--cutadapt_action", etc.
          If the 5' end adapter is longer than 5bp, it is suggest to trim the adapt unless it is a part of reference sequence.
          Please double check the 5' end adapter by top 10K R1 reads with the tool provided in this webpage.
          The input R1 reads must be a plain text file. The tool does not accept the .gz file.</li>
        <li>What is the correct genome? If wrong genome is defined, it will affect the mapping rate.</li>
        <li>What is the experiment design? The "--input" and "--output" parameters are required. See "Usage docs" to check how to prepare the input csv file.</li>
      </ol>
    </p>
    <div class="alert border bg-light small">
        <code id="code"></code>
    </div>
    <h1 id="parameters" class="">
      Parameters
      <a href="#parameters" class="header-link"><span class="fas fa-link"></span></a>
    </h1>
  <from id="cmdform">

      <div class="row param-docs-row border-bottom align-items-center">
          <div class="col-10 col-md-5 col-lg-4 col-xl-3 small-h ps-3 param-docs-row-id-col">
              <label for="profile">profile</label>
          </div>
          <div class="col me-auto">
        <select id="profile" name="profile" class="hicarinput">
          <option value="" selected></option>
          <option value="singularity">singularity</option>
          <option value="docker">docker</option>
          <option value="conda">conda</option>
          <option value="podman">podman</option>
          <option value="shifter">shifter</option>
          <option value="charliecloud">charliecloud</option>
          <option value="institute">institute</option>
        </select>
        </div>
      </div>
      <div class="row param-docs-row border-bottom align-items-center">
          <div class="col-10 col-md-5 col-lg-4 col-xl-3 small-h ps-3 param-docs-row-id-col">
              <label for="config">config</label>
          </div>
          <div class="col me-auto">
        <input type="text" id="config" name="config" value="profile.config" class="hicarinput">
          </div>
          <div class="col-auto d-flex flex-column align-items-end my-1">
            <div id="--input-body" class="param-docs-body small">
              <p>Path to profile.config file.</p>
            </div>
          </div>
      </div>
  
    <div class="row param-docs-row border-bottom align-items-baseline mt-5 param-docs-row-group ">
      <div class="col-12 param-docs-row-id-col"><h2>Input/output options</h2></div>
      <div class="col me-auto"></div>
      <div class="col-auto d-flex flex-column align-items-end my-1"></div>
    </div>
    <div class="row param-docs-row border-bottom align-items-center ">
      <div class="col-10 col-md-5 col-lg-4 col-xl-3 small-h ps-3 param-docs-row-id-col"><h3><code>--input</code></h3></div>
      <div class="col me-auto">
                  <input id="input" name="input" type="file" class="hicarinput" required>
            </div>
      <div class="col-auto d-flex flex-column align-items-end my-1"><div id="--input-body" class="param-docs-body small"><p>Path to comma-separated file containing information about the samples in the experiment.</p></div><div class="param_labels_helpbtn d-flex align-items-center">
        <button class="btn btn-sm btn-outline-info ms-2 mb-1 mt-1" data-bs-toggle="collapse" href="#input-help" aria-expanded="false">
          <i class="fas fa-question-circle"></i> Help
        </button><span class="badge bg-warning text-dark small ms-2">required</span></div></div>
        <div class="collapse bg-lightgray col-12 param-docs-help-text p-2 mb-2 small d-print-block" id="input-help">
          <p>You will need to create a design file with information about the samples in your experiment before running the pipeline. Use this parameter to specify its location. It has to be a comma-separated file with 3 columns, and a header row. See <a href="https://nf-co.re/hicar/usage#samplesheet-input">usage docs</a>.</p></div>
    </div>
    <div class="row param-docs-row border-bottom align-items-center ">
      <div class="col-10 col-md-5 col-lg-4 col-xl-3 small-h ps-3 param-docs-row-id-col"><h3><code>--method</code></h3></div>
      <div class="col me-auto">
                  <input id="method" name="method" type="text" value="HiCAR" class="hicarinput" >
              </div>
      <div class="col-auto d-flex flex-column align-items-end my-1"><div id="--method-body" class="param-docs-body small"><p>Metho for the experiment.</p></div><div class="param_labels_helpbtn d-flex align-items-center">
        <button class="btn btn-sm btn-outline-info ms-2 mb-1 mt-1" data-bs-toggle="collapse" href="#method-help" aria-expanded="false">
          <i class="fas fa-question-circle"></i> Help
        </button></div></div>
        <div class="collapse bg-lightgray col-12 param-docs-help-text p-2 mb-2 small d-print-block" id="method-help">
          <p>It can be HiCAR, HiChIP, ChIA-PET, PLAC-Seq. If it is not HiCAR, user may want to define the anchor peaks. Otherwise, the anchor peaks will be called by R2 reads.</p></div>
    </div>
    <div class="row param-docs-row border-bottom align-items-center ">
      <div class="col-10 col-md-5 col-lg-4 col-xl-3 small-h ps-3 param-docs-row-id-col"><h3><code>--anchor_peaks</code></h3></div>
      <div class="col me-auto">
                  <input id="anchor_peaks" name="anchor_peaks" type="file" class="hicarinput" >
            </div>
      <div class="col-auto d-flex flex-column align-items-end my-1"><div id="--anchor_peaks-body" class="param-docs-body small"><p>Path to anchor peaks</p></div><div class="param_labels_helpbtn d-flex align-items-center">
        <button class="btn btn-sm btn-outline-info ms-2 mb-1 mt-1" data-bs-toggle="collapse" href="#anchor_peaks-help" aria-expanded="false">
          <i class="fas fa-question-circle"></i> Help
        </button></div></div>
        <div class="collapse bg-lightgray col-12 param-docs-help-text p-2 mb-2 small d-print-block" id="anchor_peaks-help">
          <p>The anchor peaks can be a narrowPeak or broadPeak.</p></div>
    </div>
    <div class="row param-docs-row border-bottom align-items-center ">
      <div class="col-10 col-md-5 col-lg-4 col-xl-3 small-h ps-3 param-docs-row-id-col"><h3><code>--outdir</code></h3></div>
      <div class="col me-auto">
                <input id="outdir" name="outdir" type="text" value="results" class="hicarinput" minlength="2" required>
            </div>
      <div class="col-auto d-flex flex-column align-items-end my-1"><div id="--outdir-body" class="param-docs-body small"><p>The output directory where the results will be saved. You have to use absolute paths to storage on Cloud infrastructure.</p></div><div class="param_labels_helpbtn d-flex align-items-center"><span class="badge bg-warning text-dark small ms-2">required</span></div></div>
    </div>
    <div class="row param-docs-row border-bottom align-items-center ">
      <div class="col-10 col-md-5 col-lg-4 col-xl-3 small-h ps-3 param-docs-row-id-col"><h3><code>--email</code></h3></div>
      <div class="col me-auto">
                  <input id="email" name="email" type="text" value="" class="hicarinput" >
              </div>
      <div class="col-auto d-flex flex-column align-items-end my-1"><div id="--email-body" class="param-docs-body small"><p>Email address for completion summary.</p></div><div class="param_labels_helpbtn d-flex align-items-center">
        <button class="btn btn-sm btn-outline-info ms-2 mb-1 mt-1" data-bs-toggle="collapse" href="#email-help" aria-expanded="false">
          <i class="fas fa-question-circle"></i> Help
        </button></div></div>
        <div class="collapse bg-lightgray col-12 param-docs-help-text p-2 mb-2 small d-print-block" id="email-help">
          <p>Set this parameter to your e-mail address to get a summary e-mail with details of the run sent to you when the workflow exits. If set in your user config file (<code>~/.nextflow/config</code>) then you don't need to specify this on the command line for every run.</p></div>
    </div>
    <div class="row param-docs-row border-bottom align-items-center ">
      <div class="col-10 col-md-5 col-lg-4 col-xl-3 small-h ps-3 param-docs-row-id-col"><h3><code>--multiqc_title</code></h3></div>
      <div class="col me-auto">
                  <input id="multiqc_title" name="multiqc_title" type="text" value="" class="hicarinput" >
              </div>
      <div class="col-auto d-flex flex-column align-items-end my-1"><div id="--multiqc_title-body" class="param-docs-body small"><p>MultiQC report title. Printed as page header, used for filename if not otherwise specified.</p></div></div>
    </div>
    <div class="row param-docs-row border-bottom align-items-baseline mt-5 param-docs-row-group ">
      <div class="col-12 param-docs-row-id-col"><h2>Reference genome options</h2></div>
      <div class="col me-auto"></div>
      <div class="col-auto d-flex flex-column align-items-end my-1"></div>
    </div>
    <div class="row param-docs-row border-bottom align-items-center ">
      <div class="col-10 col-md-5 col-lg-4 col-xl-3 small-h ps-3 param-docs-row-id-col"><h3><code>--genome</code></h3></div>
      <div class="col me-auto">
                <select id="genome" name="genome" class="hicarinput" >
                      <option value="GRCh38">GRCh38</option>
                      <option value="GRCh37">GRCh37</option>
                      <option value="hg38">hg38</option>
                      <option value="hg19">hg19</option>
                      <option value="GRCm38">GRCm38</option>
                      <option value="mm10">mm10</option>
                      <option value="custom">custom</option>
                      <option value="WBcel235">WBcel235</option>
                      <option value="ce10">ce10</option>
                      <option value="CanFam3.1">CanFam3.1</option>
                      <option value="canFam3">canFam3</option>
                      <option value="GRCz10">GRCz10</option>
                      <option value="danRer10">danRer10</option>
                      <option value="BDGP6">BDGP6</option>
                      <option value="dm6">dm6</option>
                      <option value="EquCab2">EquCab2</option>
                      <option value="equCab2">equCab2</option>
                      <option value="Galgal4">Galgal4</option>
                      <option value="galGal4">galGal4</option>
                      <option value="Rnor_5.0">Rnor_5.0</option>
                      <option value="Rnor_6.0">Rnor_6.0</option>
                      <option value="rn6">rn6</option>
                      <option value="R64-1-1">R64-1-1</option>
                      <option value="EF2">EF2</option>
                      <option value="Sbi1">Sbi1</option>
                      <option value="Sscrofa10.2">Sscrofa10.2</option>
                      <option value="AGPv3">AGPv3</option>
                      <option value="CHM13">CHM13</option>
                      <option value="TAIR10">TAIR10</option>
                      <option value="EB1">EB1</option>
                      <option value="EB2">EB2</option>
                      <option value="UMD3.1">UMD3.1</option>
                      <option value="Gm01">Gm01</option>
                      <option value="Mmul_1">Mmul_1</option>
                      <option value="IRGSP-1.0">IRGSP-1.0</option>
                      <option value="CHIMP2.1.4">CHIMP2.1.4</option>
                      <option value="bosTau8">bosTau8</option>
                      <option value="panTro4">panTro4</option>
                      <option value="sacCer3">sacCer3</option>
                      <option value="susScr3">susScr3</option>
                    </select>
            </div>
      <div class="col-auto d-flex flex-column align-items-end my-1"><div id="--genome-body" class="param-docs-body small"><p>Name of iGenomes reference.</p></div><div class="param_labels_helpbtn d-flex align-items-center">
        <button class="btn btn-sm btn-outline-info ms-2 mb-1 mt-1" data-bs-toggle="collapse" href="#genome-help" aria-expanded="false">
          <i class="fas fa-question-circle"></i> Help
        </button></div></div>
        <div class="collapse bg-lightgray col-12 param-docs-help-text p-2 mb-2 small d-print-block" id="genome-help">
          <p>If using a reference genome configured in the pipeline using iGenomes, use this parameter to give the ID for the reference. This is then used to build the full paths for all required reference genome files e.g. <code><span class="text-nowrap">--genome</span> GRCh38</code>. </p>
<p>See the <a href="https://nf-co.re/usage/reference_genomes">nf-core website docs</a> for more details.</p></div>
    </div>
    <div class="row param-docs-row border-bottom align-items-center genomeoption">
      <div class="col-10 col-md-5 col-lg-4 col-xl-3 small-h ps-3 param-docs-row-id-col"><h3><code>--fasta</code></h3></div>
      <div class="col me-auto">
                  <input id="fasta" name="fasta" type="file" class="hicarinput" >
            </div>
      <div class="col-auto d-flex flex-column align-items-end my-1"><div id="--fasta-body" class="param-docs-body small"><p>Path to FASTA genome file.</p></div><div class="param_labels_helpbtn d-flex align-items-center">
        <button class="btn btn-sm btn-outline-info ms-2 mb-1 mt-1" data-bs-toggle="collapse" href="#fasta-help" aria-expanded="false">
          <i class="fas fa-question-circle"></i> Help
        </button></div></div>
        <div class="collapse bg-lightgray col-12 param-docs-help-text p-2 mb-2 small d-print-block" id="fasta-help">
          <p>This parameter is <em>mandatory</em> if <code><span class="text-nowrap">--genome</code></span> is not specified. If you don't have a BWA index available this will be generated for you automatically. Combine with <code><span class="text-nowrap">--save_reference</code></span> to save BWA index for future runs.</p></div>
    </div>
    <div class="row param-docs-row border-bottom align-items-center param-docs-hidden collapse d-print-block ">
      <div class="col-10 col-md-5 col-lg-4 col-xl-3 small-h ps-3 param-docs-row-id-col"><h3><code>--igenomes_base</code></h3></div>
      <div class="col me-auto"></div>
      <div class="col-auto d-flex flex-column align-items-end my-1"><div id="--igenomes_base-body" class="param-docs-body small"><p>Directory / URL base for iGenomes references.</p></div><div class="param_labels_helpbtn d-flex align-items-center"><span class="badge bg-light text-dark small ms-2">hidden</span></div></div>
    </div>
    <div class="row param-docs-row border-bottom align-items-center param-docs-hidden collapse d-print-block ">
      <div class="col-10 col-md-5 col-lg-4 col-xl-3 small-h ps-3 param-docs-row-id-col"><h3><code>--igenomes_ignore</code></h3></div>
      <div class="col me-auto"></div>
      <div class="col-auto d-flex flex-column align-items-end my-1"><div id="--igenomes_ignore-body" class="param-docs-body small"><p>Do not load the iGenomes reference config.</p></div><div class="param_labels_helpbtn d-flex align-items-center">
        <button class="btn btn-sm btn-outline-info ms-2 mb-1 mt-1" data-bs-toggle="collapse" href="#igenomes_ignore-help" aria-expanded="false">
          <i class="fas fa-question-circle"></i> Help
        </button><span class="badge bg-light text-dark small ms-2">hidden</span></div></div>
        <div class="collapse bg-lightgray col-12 param-docs-help-text p-2 mb-2 small d-print-block" id="igenomes_ignore-help">
          <p>Do not load <code>igenomes.config</code> when running the pipeline. You may choose this option if you observe clashes between custom parameters and those supplied in <code>igenomes.config</code>.</p></div>
    </div>
    <div class="row param-docs-row border-bottom align-items-center genomeoption">
      <div class="col-10 col-md-5 col-lg-4 col-xl-3 small-h ps-3 param-docs-row-id-col"><h3><code>--bwa_index</code></h3></div>
      <div class="col me-auto">
                  <input id="bwa_index" name="bwa_index" type="text" value="" class="hicarinput" >
              </div>
      <div class="col-auto d-flex flex-column align-items-end my-1"><div id="--bwa_index-body" class="param-docs-body small"><p>Path to bwa index file.</p></div><div class="param_labels_helpbtn d-flex align-items-center">
        <button class="btn btn-sm btn-outline-info ms-2 mb-1 mt-1" data-bs-toggle="collapse" href="#bwa_index-help" aria-expanded="false">
          <i class="fas fa-question-circle"></i> Help
        </button></div></div>
        <div class="collapse bg-lightgray col-12 param-docs-help-text p-2 mb-2 small d-print-block" id="bwa_index-help">
          <p>If you have no genome reference available, the pipeline can build one using a FASTA file. This requires additional time and resources, so it's better to use a pre-build index if possible.</p></div>
    </div>
    <div class="row param-docs-row border-bottom align-items-center genomeoption">
      <div class="col-10 col-md-5 col-lg-4 col-xl-3 small-h ps-3 param-docs-row-id-col"><h3><code>--gtf</code></h3></div>
      <div class="col me-auto">
                  <input id="gtf" name="gtf" type="file" class="hicarinput" >
            </div>
      <div class="col-auto d-flex flex-column align-items-end my-1"><div id="--gtf-body" class="param-docs-body small"><p>Path to annotation gtf file.</p></div><div class="param_labels_helpbtn d-flex align-items-center">
        <button class="btn btn-sm btn-outline-info ms-2 mb-1 mt-1" data-bs-toggle="collapse" href="#gtf-help" aria-expanded="false">
          <i class="fas fa-question-circle"></i> Help
        </button></div></div>
        <div class="collapse bg-lightgray col-12 param-docs-help-text p-2 mb-2 small d-print-block" id="gtf-help">
          <p>Annotation gtf or gff file is required for annotation and quality control. This parameter is <em>mandatory</em> if <code><span class="text-nowrap">--genome</code></span> is not specified.</p></div>
    </div>
    <div class="row param-docs-row border-bottom align-items-center genomeoption">
      <div class="col-10 col-md-5 col-lg-4 col-xl-3 small-h ps-3 param-docs-row-id-col"><h3><code>--gff</code></h3></div>
      <div class="col me-auto">
                  <input id="gff" name="gff" type="file" class="hicarinput" >
            </div>
      <div class="col-auto d-flex flex-column align-items-end my-1"><div id="--gff-body" class="param-docs-body small"><p>Path to annotation gff file.</p></div><div class="param_labels_helpbtn d-flex align-items-center">
        <button class="btn btn-sm btn-outline-info ms-2 mb-1 mt-1" data-bs-toggle="collapse" href="#gff-help" aria-expanded="false">
          <i class="fas fa-question-circle"></i> Help
        </button></div></div>
        <div class="collapse bg-lightgray col-12 param-docs-help-text p-2 mb-2 small d-print-block" id="gff-help">
          <p>Annotation gff or gff file is required for annotation and quality control. This parameter is <em>mandatory</em> if <code><span class="text-nowrap">--genome</code></span> is not specified.</p></div>
    </div>
    <div class="row param-docs-row border-bottom align-items-center genomeoption">
      <div class="col-10 col-md-5 col-lg-4 col-xl-3 small-h ps-3 param-docs-row-id-col"><h3><code>--gene_bed</code></h3></div>
      <div class="col me-auto">
                  <input id="gene_bed" name="gene_bed" type="file" class="hicarinput" >
            </div>
      <div class="col-auto d-flex flex-column align-items-end my-1"><div id="--gene_bed-body" class="param-docs-body small"><p>Path to annotation gene bed file.</p></div><div class="param_labels_helpbtn d-flex align-items-center">
        <button class="btn btn-sm btn-outline-info ms-2 mb-1 mt-1" data-bs-toggle="collapse" href="#gene_bed-help" aria-expanded="false">
          <i class="fas fa-question-circle"></i> Help
        </button></div></div>
        <div class="collapse bg-lightgray col-12 param-docs-help-text p-2 mb-2 small d-print-block" id="gene_bed-help">
          <p>Path to BED file containing gene intervals. This will be created from the GTF file if not specified.</p></div>
    </div>
    <div class="row param-docs-row border-bottom align-items-center genomeoption">
      <div class="col-10 col-md-5 col-lg-4 col-xl-3 small-h ps-3 param-docs-row-id-col"><h3><code>--mappability</code></h3></div>
      <div class="col me-auto">
                  <input id="mappability" name="mappability" type="file" class="hicarinput" >
            </div>
      <div class="col-auto d-flex flex-column align-items-end my-1"><div id="--mappability-body" class="param-docs-body small"><p>Path to genome mappability file.</p></div><div class="param_labels_helpbtn d-flex align-items-center">
        <button class="btn btn-sm btn-outline-info ms-2 mb-1 mt-1" data-bs-toggle="collapse" href="#mappability-help" aria-expanded="false">
          <i class="fas fa-question-circle"></i> Help
        </button></div></div>
        <div class="collapse bg-lightgray col-12 param-docs-help-text p-2 mb-2 small d-print-block" id="mappability-help">
          <p>If you have no genome mappability available, the pipeline can build one using a FASTA file. This requires additional time and resources, so it's better to use a pre-build index if possible.</p></div>
    </div>
    <div class="row param-docs-row border-bottom align-items-center genomeoption">
      <div class="col-10 col-md-5 col-lg-4 col-xl-3 small-h ps-3 param-docs-row-id-col"><h3><code>--macs_gsize</code></h3></div>
      <div class="col me-auto">
                  <input id="macs_gsize" name="macs_gsize" type="text" value="" class="hicarinput" >
              </div>
      <div class="col-auto d-flex flex-column align-items-end my-1"><div id="--macs_gsize-body" class="param-docs-body small"><p>Effective genome size parameter required by MACS2.</p></div><div class="param_labels_helpbtn d-flex align-items-center">
        <button class="btn btn-sm btn-outline-info ms-2 mb-1 mt-1" data-bs-toggle="collapse" href="#macs_gsize-help" aria-expanded="false">
          <i class="fas fa-question-circle"></i> Help
        </button></div></div>
        <div class="collapse bg-lightgray col-12 param-docs-help-text p-2 mb-2 small d-print-block" id="macs_gsize-help">
          <p><a href="https://github.com/taoliu/MACS#-g--gsize">Effective genome size</a> parameter required by MACS2. If using an iGenomes reference these have been provided when <code><span class="text-nowrap">--genome</code></span> is set as <em>GRCh37</em>, <em>GRCh38</em>, <em>GRCm38</em>, <em>WBcel235</em>, <em>BDGP6</em>, <em>R64-1-1</em>, <em>EF2</em>, <em>hg38</em>, <em>hg19</em> and <em>mm10</em>. For other genomes, if this parameter is not specified then the MACS2 peak-calling and differential analysis will be skipped.</p></div>
    </div>
    <div class="row param-docs-row border-bottom align-items-center genomeoption">
      <div class="col-10 col-md-5 col-lg-4 col-xl-3 small-h ps-3 param-docs-row-id-col"><h3><code>--ucscname</code></h3></div>
      <div class="col me-auto">
                  <input id="ucscname" name="ucscname" type="text" value="" class="hicarinput" >
              </div>
      <div class="col-auto d-flex flex-column align-items-end my-1"><div id="--ucscname-body" class="param-docs-body small"><p>UCSC assembly annotation name.</p></div><div class="param_labels_helpbtn d-flex align-items-center">
        <button class="btn btn-sm btn-outline-info ms-2 mb-1 mt-1" data-bs-toggle="collapse" href="#ucscname-help" aria-expanded="false">
          <i class="fas fa-question-circle"></i> Help
        </button></div></div>
        <div class="collapse bg-lightgray col-12 param-docs-help-text p-2 mb-2 small d-print-block" id="ucscname-help">
          <p><a href="https://genome.ucsc.edu/FAQ/FAQgenes.html">UCSC assembly annotation name</a></p></div>
    </div>
    <div class="row param-docs-row border-bottom align-items-center genomeoption">
      <div class="col-10 col-md-5 col-lg-4 col-xl-3 small-h ps-3 param-docs-row-id-col"><h3><code>--blacklist</code></h3></div>
      <div class="col me-auto">
                  <input id="blacklist" name="blacklist" type="file" class="hicarinput" >
            </div>
      <div class="col-auto d-flex flex-column align-items-end my-1"><div id="--blacklist-body" class="param-docs-body small"><p>Path to blacklist regions in BED format, used for filtering alignments.</p></div><div class="param_labels_helpbtn d-flex align-items-center">
        <button class="btn btn-sm btn-outline-info ms-2 mb-1 mt-1" data-bs-toggle="collapse" href="#blacklist-help" aria-expanded="false">
          <i class="fas fa-question-circle"></i> Help
        </button></div></div>
        <div class="collapse bg-lightgray col-12 param-docs-help-text p-2 mb-2 small d-print-block" id="blacklist-help">
          <p>If provided, alignments that overlap with the regions in this file will be filtered out (see <a href="https://sites.google.com/site/anshulkundaje/projects/blacklists">ENCODE blacklists</a>). The file should be in BED format. Blacklisted regions for <em>GRCh37</em>, <em>GRCh38</em>, <em>GRCm38</em>, <em>hg19</em>, <em>hg38</em>, <em>mm10</em> are bundled with the pipeline in the <a href="../assets/blacklists/"><code>blacklists</code></a> directory, and as such will be automatically used if any of those genomes are specified with the <code><span class="text-nowrap">--genome</code></span> parameter.</p></div>
    </div>
    <div class="row param-docs-row border-bottom align-items-center ">
      <div class="col-10 col-md-5 col-lg-4 col-xl-3 small-h ps-3 param-docs-row-id-col"><h3><code>--publish_mappability</code></h3></div>
      <div class="col me-auto">
                  <input id="publish_mappability" name="publish_mappability" type="checkbox" class="hicarinput" >
              </div>
      <div class="col-auto d-flex flex-column align-items-end my-1"><div id="--publish_mappability-body" class="param-docs-body small"><p>publish mappability to results genome/mappability folder</p></div></div>
    </div>
    <div class="row param-docs-row border-bottom align-items-center ">
      <div class="col-10 col-md-5 col-lg-4 col-xl-3 small-h ps-3 param-docs-row-id-col"><h3><code>--publish_genome</code></h3></div>
      <div class="col me-auto">
                  <input id="publish_genome" name="publish_genome" type="checkbox" class="hicarinput" >
              </div>
      <div class="col-auto d-flex flex-column align-items-end my-1"><div id="--publish_genome-body" class="param-docs-body small"><p>publish genome</p></div></div>
    </div>
    <div class="row param-docs-row border-bottom align-items-baseline mt-5 param-docs-row-group ">
      <div class="col-12 param-docs-row-id-col"><h2>Experiment design options</h2></div>
      <div class="col me-auto"></div>
      <div class="col-auto d-flex flex-column align-items-end my-1"></div>
    </div>
    <div class="row param-docs-row border-bottom align-items-center ">
      <div class="col-10 col-md-5 col-lg-4 col-xl-3 small-h ps-3 param-docs-row-id-col"><h3><code>--enzyme</code></h3></div>
      <div class="col me-auto">
                  <input id="enzyme" name="enzyme" type="text" value="CviQI" class="hicarinput" >
              </div>
      <div class="col-auto d-flex flex-column align-items-end my-1"><div id="--enzyme-body" class="param-docs-body small"><p>Specifies that the cutting position has to be using.</p></div><div class="param_labels_helpbtn d-flex align-items-center">
        <button class="btn btn-sm btn-outline-info ms-2 mb-1 mt-1" data-bs-toggle="collapse" href="#enzyme-help" aria-expanded="false">
          <i class="fas fa-question-circle"></i> Help
        </button></div></div>
        <div class="collapse bg-lightgray col-12 param-docs-help-text p-2 mb-2 small d-print-block" id="enzyme-help">
          <p>Default CviQI digestion. Available enzymes are MboI, DpnII, BglII, HindIII, MseI, and CviQI.</p></div>
    </div>
    <div class="row param-docs-row border-bottom align-items-center param-docs-hidden collapse d-print-block ">
      <div class="col-10 col-md-5 col-lg-4 col-xl-3 small-h ps-3 param-docs-row-id-col"><h3><code>--restriction_sites</code></h3></div>
      <div class="col me-auto"></div>
      <div class="col-auto d-flex flex-column align-items-end my-1"><div id="--restriction_sites-body" class="param-docs-body small"><p>Specifies that the cutting sequence has to be using.</p></div><div class="param_labels_helpbtn d-flex align-items-center">
        <button class="btn btn-sm btn-outline-info ms-2 mb-1 mt-1" data-bs-toggle="collapse" href="#restriction_sites-help" aria-expanded="false">
          <i class="fas fa-question-circle"></i> Help
        </button><span class="badge bg-light text-dark small ms-2">hidden</span></div></div>
        <div class="collapse bg-lightgray col-12 param-docs-help-text p-2 mb-2 small d-print-block" id="restriction_sites-help">
          <p>It will be automatically assigned by the enzyme inputed.</p></div>
    </div>
    <div class="row param-docs-row border-bottom align-items-center ">
      <div class="col-10 col-md-5 col-lg-4 col-xl-3 small-h ps-3 param-docs-row-id-col"><h3><code>--restriction_sites_cut_off</code></h3></div>
      <div class="col me-auto">
                  <input id="restriction_sites_cut_off" name="restriction_sites_cut_off" type="text" value="0.5" class="hicarinput" >
              </div>
      <div class="col-auto d-flex flex-column align-items-end my-1"><div id="--restriction_sites_cut_off-body" class="param-docs-body small"><p>Specifies that the cutoff value used for mappability filter.</p></div><div class="param_labels_helpbtn d-flex align-items-center">
        <button class="btn btn-sm btn-outline-info ms-2 mb-1 mt-1" data-bs-toggle="collapse" href="#restriction_sites_cut_off-help" aria-expanded="false">
          <i class="fas fa-question-circle"></i> Help
        </button></div></div>
        <div class="collapse bg-lightgray col-12 param-docs-help-text p-2 mb-2 small d-print-block" id="restriction_sites_cut_off-help">
          <p>Default is 0.5. It is the average over just covered bases.</p></div>
    </div>
    <div class="row param-docs-row border-bottom align-items-center ">
      <div class="col-10 col-md-5 col-lg-4 col-xl-3 small-h ps-3 param-docs-row-id-col"><h3><code>--cutadapt_5end</code></h3></div>
      <div class="col me-auto">
                  <input id="cutadapt_5end" name="cutadapt_5end" type="text" value="^TAC" class="hicarinput" >
              </div>
      <div class="col-auto d-flex flex-column align-items-end my-1"><div id="--cutadapt_5end-body" class="param-docs-body small"><p>trim 5' end with the string if not skip_cutadapt</p></div></div>
         <br/>
         <div class="col-10 col-md-5 col-lg-4 col-xl-3 small-h ps-3 param-docs-row-id-col">
             <label>&nbsp;</label>
         </div>
         <div class="col-10 col-md-5 col-lg-4 col-xl-3 small-h ps-3 param-docs-row-id-col">
             <h3><code>Validate by one R1 reads</code></h3>
         </div>
         <div class="col me-auto">
                <input id="fastq_r1" name="fastq_r1" type="file"/>
         </div>
      
    </div>
    <div class="row param-docs-row border-bottom align-items-center ">
      <div class="col-10 col-md-5 col-lg-4 col-xl-3 small-h ps-3 param-docs-row-id-col"><h3><code>--cutadapt_action</code></h3></div>
      <div class="col me-auto">
              <select id="cutadapt_action" name="cutadapt_action" class="hicarinput" >
        <option value="trim" >trim</option><option value="retain" >retain</option><option value="none" selected>none</option><option value="mask" >mask</option>
              </select>
        </div>
      <div class="col-auto d-flex flex-column align-items-end my-1"><div id="--cutadapt_action-body" class="param-docs-body small"><p>The action when an adapter is found if not skip_cutadapt</p></div></div>
    </div>
    <div class="row param-docs-row border-bottom align-items-center ">
      <div class="col-10 col-md-5 col-lg-4 col-xl-3 small-h ps-3 param-docs-row-id-col"><h3><code>--cutadapt_error_tolerance</code></h3></div>
      <div class="col me-auto">
                  <input id="cutadapt_error_tolerance" name="cutadapt_error_tolerance" type="text" value="0.15" class="hicarinput" >
              </div>
      <div class="col-auto d-flex flex-column align-items-end my-1"><div id="--cutadapt_error_tolerance-body" class="param-docs-body small"><p>Error tolerance for adaptor filter if not skip_cutadapt</p></div></div>
    </div>
    <div class="row param-docs-row border-bottom align-items-baseline mt-5 param-docs-row-group ">
      <div class="col-12 param-docs-row-id-col"><h2>MACS2 peak calling options</h2></div>
      <div class="col me-auto"></div>
      <div class="col-auto d-flex flex-column align-items-end my-1"></div>
    </div>
    <div class="row param-docs-row border-bottom align-items-center ">
      <div class="col-10 col-md-5 col-lg-4 col-xl-3 small-h ps-3 param-docs-row-id-col"><h3><code>--shiftsize</code></h3></div>
      <div class="col me-auto">
                  <input id="shiftsize" name="shiftsize" type="text" value="-75" class="hicarinput" >
              </div>
      <div class="col-auto d-flex flex-column align-items-end my-1"><div id="--shiftsize-body" class="param-docs-body small"><p>shift size for MACS2</p></div><div class="param_labels_helpbtn d-flex align-items-center">
        <button class="btn btn-sm btn-outline-info ms-2 mb-1 mt-1" data-bs-toggle="collapse" href="#shiftsize-help" aria-expanded="false">
          <i class="fas fa-question-circle"></i> Help
        </button></div></div>
        <div class="collapse bg-lightgray col-12 param-docs-help-text p-2 mb-2 small d-print-block" id="shiftsize-help">
          <p>How reads will be shift to increase the peak/noise ratios. See help from <a href="https://github.com/macs3-project/MACS/blob/master/docs/callpeak.md">MACS2 documentation</a></p></div>
    </div>
    <div class="row param-docs-row border-bottom align-items-center ">
      <div class="col-10 col-md-5 col-lg-4 col-xl-3 small-h ps-3 param-docs-row-id-col"><h3><code>--smooth_window</code></h3></div>
      <div class="col me-auto">
                  <input id="smooth_window" name="smooth_window" type="text" value="150" class="hicarinput" >
              </div>
      <div class="col-auto d-flex flex-column align-items-end my-1"><div id="--smooth_window-body" class="param-docs-body small"><p>extsize for MACS2</p></div><div class="param_labels_helpbtn d-flex align-items-center">
        <button class="btn btn-sm btn-outline-info ms-2 mb-1 mt-1" data-bs-toggle="collapse" href="#smooth_window-help" aria-expanded="false">
          <i class="fas fa-question-circle"></i> Help
        </button></div></div>
        <div class="collapse bg-lightgray col-12 param-docs-help-text p-2 mb-2 small d-print-block" id="smooth_window-help">
          <p>How reads will be extend in 5'-&gt;3' direction to increase the fragment size. See help from <a href="https://github.com/macs3-project/MACS/blob/master/docs/callpeak.md">MACS2 documentation</a></p></div>
    </div>
    <div class="row param-docs-row border-bottom align-items-center ">
      <div class="col-10 col-md-5 col-lg-4 col-xl-3 small-h ps-3 param-docs-row-id-col"><h3><code>--qval_thresh</code></h3></div>
      <div class="col me-auto">
                  <input id="qval_thresh" name="qval_thresh" type="text" value="0.01" class="hicarinput" >
              </div>
      <div class="col-auto d-flex flex-column align-items-end my-1"><div id="--qval_thresh-body" class="param-docs-body small"><p>cutoff qvalue</p></div><div class="param_labels_helpbtn d-flex align-items-center">
        <button class="btn btn-sm btn-outline-info ms-2 mb-1 mt-1" data-bs-toggle="collapse" href="#qval_thresh-help" aria-expanded="false">
          <i class="fas fa-question-circle"></i> Help
        </button></div></div>
        <div class="collapse bg-lightgray col-12 param-docs-help-text p-2 mb-2 small d-print-block" id="qval_thresh-help">
          <p>The minimum FDR cutoff to call significant peak for R2 reads. See help from <a href="https://github.com/macs3-project/MACS/blob/master/docs/callpeak.md">MACS2 documentation</a></p></div>
    </div>
    <div class="row param-docs-row border-bottom align-items-baseline mt-5 param-docs-row-group ">
      <div class="col-12 param-docs-row-id-col"><h2>Interactions calling options</h2></div>
      <div class="col me-auto"></div>
      <div class="col-auto d-flex flex-column align-items-end my-1"></div>
    </div>
    <div class="row param-docs-row border-bottom align-items-center ">
      <div class="col-10 col-md-5 col-lg-4 col-xl-3 small-h ps-3 param-docs-row-id-col"><h3><code>--cool_bin</code></h3></div>
      <div class="col me-auto">
                  <input id="cool_bin" name="cool_bin" type="text" value="5000_10000" class="hicarinput" >
              </div>
      <div class="col-auto d-flex flex-column align-items-end my-1"><div id="--cool_bin-body" class="param-docs-body small"><p>resolution bin size</p></div><div class="param_labels_helpbtn d-flex align-items-center">
        <button class="btn btn-sm btn-outline-info ms-2 mb-1 mt-1" data-bs-toggle="collapse" href="#cool_bin-help" aria-expanded="false">
          <i class="fas fa-question-circle"></i> Help
        </button></div></div>
        <div class="collapse bg-lightgray col-12 param-docs-help-text p-2 mb-2 small d-print-block" id="cool_bin-help">
          <p>Bin size for singal resampling.</p></div>
    </div>
    <div class="row param-docs-row border-bottom align-items-center ">
      <div class="col-10 col-md-5 col-lg-4 col-xl-3 small-h ps-3 param-docs-row-id-col"><h3><code>--maps_digest_file</code></h3></div>
      <div class="col me-auto">
                  <input id="maps_digest_file" name="maps_digest_file" type="text" value="None" class="hicarinput" >
              </div>
      <div class="col-auto d-flex flex-column align-items-end my-1"><div id="--maps_digest_file-body" class="param-docs-body small"><p>output of restriction_cut_multipleenzyme.py.</p></div><div class="param_labels_helpbtn d-flex align-items-center">
        <button class="btn btn-sm btn-outline-info ms-2 mb-1 mt-1" data-bs-toggle="collapse" href="#maps_digest_file-help" aria-expanded="false">
          <i class="fas fa-question-circle"></i> Help
        </button></div></div>
        <div class="collapse bg-lightgray col-12 param-docs-help-text p-2 mb-2 small d-print-block" id="maps_digest_file-help">
          <p>See <a href="https://github.com/ijuric/MAPS/blob/master/bin/utils/genomic_features_generator/scripts/restriction_cut_multipleenzyme.py">https://github.com/ijuric/MAPS/blob/master/bin/utils/genomic_features_generator/scripts/restriction_cut_multipleenzyme.py</a>.</p></div>
    </div>
    <div class="row param-docs-row border-bottom align-items-center ">
      <div class="col-10 col-md-5 col-lg-4 col-xl-3 small-h ps-3 param-docs-row-id-col"><h3><code>--maps_cutoff_counts</code></h3></div>
      <div class="col me-auto">
                  <input id="maps_cutoff_counts" name="maps_cutoff_counts" type="text" value="12" class="hicarinput" >
              </div>
      <div class="col-auto d-flex flex-column align-items-end my-1"><div id="--maps_cutoff_counts-body" class="param-docs-body small"><p>MAPS regression cutoff value</p></div><div class="param_labels_helpbtn d-flex align-items-center">
        <button class="btn btn-sm btn-outline-info ms-2 mb-1 mt-1" data-bs-toggle="collapse" href="#maps_cutoff_counts-help" aria-expanded="false">
          <i class="fas fa-question-circle"></i> Help
        </button></div></div>
        <div class="collapse bg-lightgray col-12 param-docs-help-text p-2 mb-2 small d-print-block" id="maps_cutoff_counts-help">
          <p>Minimal cutoff value for coverage in the called connections. See help from <a href="https://github.com/ijuric/MAPS">MAPS documentation</a>, default 12.</p></div>
    </div>
    <div class="row param-docs-row border-bottom align-items-center ">
      <div class="col-10 col-md-5 col-lg-4 col-xl-3 small-h ps-3 param-docs-row-id-col"><h3><code>--maps_cutoff_fold_change</code></h3></div>
      <div class="col me-auto">
                  <input id="maps_cutoff_fold_change" name="maps_cutoff_fold_change" type="text" value="2" class="hicarinput" >
              </div>
      <div class="col-auto d-flex flex-column align-items-end my-1"><div id="--maps_cutoff_fold_change-body" class="param-docs-body small"><p>MAPS regression fold change cutoff value</p></div><div class="param_labels_helpbtn d-flex align-items-center">
        <button class="btn btn-sm btn-outline-info ms-2 mb-1 mt-1" data-bs-toggle="collapse" href="#maps_cutoff_fold_change-help" aria-expanded="false">
          <i class="fas fa-question-circle"></i> Help
        </button></div></div>
        <div class="collapse bg-lightgray col-12 param-docs-help-text p-2 mb-2 small d-print-block" id="maps_cutoff_fold_change-help">
          <p>Miniaml cutoff value for fold change from signal to expected values. See help from <a href="https://github.com/ijuric/MAPS">MAPS documentation</a>, default 2.</p></div>
    </div>
    <div class="row param-docs-row border-bottom align-items-center ">
      <div class="col-10 col-md-5 col-lg-4 col-xl-3 small-h ps-3 param-docs-row-id-col"><h3><code>--maps_cutoff_fdr</code></h3></div>
      <div class="col me-auto">
                  <input id="maps_cutoff_fdr" name="maps_cutoff_fdr" type="text" value="2" class="hicarinput" >
              </div>
      <div class="col-auto d-flex flex-column align-items-end my-1"><div id="--maps_cutoff_fdr-body" class="param-docs-body small"><p>MAPS regression -log10(fdr) cutoff value</p></div><div class="param_labels_helpbtn d-flex align-items-center">
        <button class="btn btn-sm btn-outline-info ms-2 mb-1 mt-1" data-bs-toggle="collapse" href="#maps_cutoff_fdr-help" aria-expanded="false">
          <i class="fas fa-question-circle"></i> Help
        </button></div></div>
        <div class="collapse bg-lightgray col-12 param-docs-help-text p-2 mb-2 small d-print-block" id="maps_cutoff_fdr-help">
          <p>Minimal cutoff -log10(FDR) value in the called connections. See help from <a href="https://github.com/ijuric/MAPS">MAPS documentation</a>, default 2 (=fdr 0.01).</p></div>
    </div>
    <div class="row param-docs-row border-bottom align-items-center ">
      <div class="col-10 col-md-5 col-lg-4 col-xl-3 small-h ps-3 param-docs-row-id-col"><h3><code>--maps_filter</code></h3></div>
      <div class="col me-auto">
                  <input id="maps_filter" name="maps_filter" type="text" value="None" class="hicarinput" >
              </div>
      <div class="col-auto d-flex flex-column align-items-end my-1"><div id="--maps_filter-body" class="param-docs-body small"><p>MAPS regression filter file name</p></div><div class="param_labels_helpbtn d-flex align-items-center">
        <button class="btn btn-sm btn-outline-info ms-2 mb-1 mt-1" data-bs-toggle="collapse" href="#maps_filter-help" aria-expanded="false">
          <i class="fas fa-question-circle"></i> Help
        </button></div></div>
        <div class="collapse bg-lightgray col-12 param-docs-help-text p-2 mb-2 small d-print-block" id="maps_filter-help">
          <p>file containing bins that need to be filtered out. Format: two columns 'chrom', 'bin'. 'chrom' contains 'chr1','chr2',.. 'bin' is bin label</p></div>
    </div>
    <div class="row param-docs-row border-bottom align-items-center ">
      <div class="col-10 col-md-5 col-lg-4 col-xl-3 small-h ps-3 param-docs-row-id-col"><h3><code>--maps_model</code></h3></div>
      <div class="col me-auto">
              <select id="maps_model" name="maps_model" class="hicarinput" >
        <option value="pospoisson" selected>pospoisson</option><option value="negbinom" >negbinom</option>
              </select>
        </div>
      <div class="col-auto d-flex flex-column align-items-end my-1"><div id="--maps_model-body" class="param-docs-body small"><p>MAPS regression type</p></div><div class="param_labels_helpbtn d-flex align-items-center">
        <button class="btn btn-sm btn-outline-info ms-2 mb-1 mt-1" data-bs-toggle="collapse" href="#maps_model-help" aria-expanded="false">
          <i class="fas fa-question-circle"></i> Help
        </button></div></div>
        <div class="collapse bg-lightgray col-12 param-docs-help-text p-2 mb-2 small d-print-block" id="maps_model-help">
          <p>pospoisson for positive poisson regression, negbinom for negative binomial. default is pospoisson.</p></div>
    </div>
    <div class="row param-docs-row border-bottom align-items-center param-docs-hidden collapse d-print-block ">
      <div class="col-10 col-md-5 col-lg-4 col-xl-3 small-h ps-3 param-docs-row-id-col"><h3><code>--remove_dup</code></h3></div>
      <div class="col me-auto"></div>
      <div class="col-auto d-flex flex-column align-items-end my-1"><div id="--remove_dup-body" class="param-docs-body small"><p>remove duplicates for high resolution peaks or not</p></div><div class="param_labels_helpbtn d-flex align-items-center"><span class="badge bg-light text-dark small ms-2">hidden</span></div></div>
    </div>
    <div class="row param-docs-row border-bottom align-items-center ">
      <div class="col-10 col-md-5 col-lg-4 col-xl-3 small-h ps-3 param-docs-row-id-col"><h3><code>--snow_type</code></h3></div>
      <div class="col me-auto">
                  <input id="snow_type" name="snow_type" type="text" value="SOCK" class="hicarinput" >
              </div>
      <div class="col-auto d-flex flex-column align-items-end my-1"><div id="--snow_type-body" class="param-docs-body small"><p>Type of snow cluster to use</p></div><div class="param_labels_helpbtn d-flex align-items-center">
        <button class="btn btn-sm btn-outline-info ms-2 mb-1 mt-1" data-bs-toggle="collapse" href="#snow_type-help" aria-expanded="false">
          <i class="fas fa-question-circle"></i> Help
        </button></div></div>
        <div class="collapse bg-lightgray col-12 param-docs-help-text p-2 mb-2 small d-print-block" id="snow_type-help">
          <p>Possible values are 'SOCK' (default), 'MPI' and 'FORK'. This will be used by snow package for HiPeak calling</p></div>
    </div>
    <div class="row param-docs-row border-bottom align-items-center param-docs-hidden collapse d-print-block ">
      <div class="col-10 col-md-5 col-lg-4 col-xl-3 small-h ps-3 param-docs-row-id-col"><h3><code>--peak_pair_block</code></h3></div>
      <div class="col me-auto"></div>
      <div class="col-auto d-flex flex-column align-items-end my-1"><div id="--peak_pair_block-body" class="param-docs-body small"><p>The block number of peak pair</p></div><div class="param_labels_helpbtn d-flex align-items-center">
        <button class="btn btn-sm btn-outline-info ms-2 mb-1 mt-1" data-bs-toggle="collapse" href="#peak_pair_block-help" aria-expanded="false">
          <i class="fas fa-question-circle"></i> Help
        </button><span class="badge bg-light text-dark small ms-2">hidden</span></div></div>
        <div class="collapse bg-lightgray col-12 param-docs-help-text p-2 mb-2 small d-print-block" id="peak_pair_block-help">
          <p>The block number of peak pair in each thread to count the reads. Smaller number will decrease the memory cost and increase the computation time for parepare the counts in hipeak calling.</p></div>
    </div>
    <div class="row param-docs-row border-bottom align-items-center param-docs-hidden collapse d-print-block ">
      <div class="col-10 col-md-5 col-lg-4 col-xl-3 small-h ps-3 param-docs-row-id-col"><h3><code>--min_counts_per_loop</code></h3></div>
      <div class="col me-auto"></div>
      <div class="col-auto d-flex flex-column align-items-end my-1"><div id="--min_counts_per_loop-body" class="param-docs-body small"><p>The minimal number reads count of peak pair</p></div><div class="param_labels_helpbtn d-flex align-items-center"><span class="badge bg-light text-dark small ms-2">hidden</span></div></div>
    </div>
    <div class="row param-docs-row border-bottom align-items-center param-docs-hidden collapse d-print-block ">
      <div class="col-10 col-md-5 col-lg-4 col-xl-3 small-h ps-3 param-docs-row-id-col"><h3><code>--create_maps_signal</code></h3></div>
      <div class="col me-auto"></div>
      <div class="col-auto d-flex flex-column align-items-end my-1"><div id="--create_maps_signal-body" class="param-docs-body small"><p>create cooler files for MAPS or not</p></div><div class="param_labels_helpbtn d-flex align-items-center"><span class="badge bg-light text-dark small ms-2">hidden</span></div></div>
    </div>
    <div class="row param-docs-row border-bottom align-items-center ">
      <div class="col-10 col-md-5 col-lg-4 col-xl-3 small-h ps-3 param-docs-row-id-col"><h3><code>--hicdcplus_cutoff_fdr</code></h3></div>
      <div class="col me-auto">
                  <input id="hicdcplus_cutoff_fdr" name="hicdcplus_cutoff_fdr" type="text" value="0.05" class="hicarinput" >
              </div>
      <div class="col-auto d-flex flex-column align-items-end my-1"><div id="--hicdcplus_cutoff_fdr-body" class="param-docs-body small"><p>FDR cutoff value</p></div><div class="param_labels_helpbtn d-flex align-items-center">
        <button class="btn btn-sm btn-outline-info ms-2 mb-1 mt-1" data-bs-toggle="collapse" href="#hicdcplus_cutoff_fdr-help" aria-expanded="false">
          <i class="fas fa-question-circle"></i> Help
        </button></div></div>
        <div class="collapse bg-lightgray col-12 param-docs-help-text p-2 mb-2 small d-print-block" id="hicdcplus_cutoff_fdr-help">
          <p>Minimal cutoff FDR value in the called connections. See help from <a href="https://bioconductor.org/packages/HiCDCPlus/">HiCDCPlus documentation</a>, default 0.05.</p></div>
    </div>
    <div class="row param-docs-row border-bottom align-items-center ">
      <div class="col-10 col-md-5 col-lg-4 col-xl-3 small-h ps-3 param-docs-row-id-col"><h3><code>--peakachu_pretrained_url</code></h3></div>
      <div class="col me-auto">
                  <input id="peakachu_pretrained_url" name="peakachu_pretrained_url" type="text" value="http://3dgenome.fsm.northwestern.edu/peakachu/HiCAR-models/HiCAR-peakachu-pretrained." class="hicarinput" >
              </div>
      <div class="col-auto d-flex flex-column align-items-end my-1"><div id="--peakachu_pretrained_url-body" class="param-docs-body small"><p>The predefined modle download prefix string</p></div><div class="param_labels_helpbtn d-flex align-items-center">
        <button class="btn btn-sm btn-outline-info ms-2 mb-1 mt-1" data-bs-toggle="collapse" href="#peakachu_pretrained_url-help" aria-expanded="false">
          <i class="fas fa-question-circle"></i> Help
        </button></div></div>
        <div class="collapse bg-lightgray col-12 param-docs-help-text p-2 mb-2 small d-print-block" id="peakachu_pretrained_url-help">
          <p>This method is only available for HiCAR data.</p></div>
    </div>
    <div class="row param-docs-row border-bottom align-items-center ">
      <div class="col-10 col-md-5 col-lg-4 col-xl-3 small-h ps-3 param-docs-row-id-col"><h3><code>--call_high_peak</code></h3></div>
      <div class="col me-auto">
                  <input id="call_high_peak" name="call_high_peak" type="checkbox" class="hicarinput" >
              </div>
      <div class="col-auto d-flex flex-column align-items-end my-1"><div id="--call_high_peak-body" class="param-docs-body small"><p>Call high resolution peaks or not</p></div></div>
    </div>
    <div class="row param-docs-row border-bottom align-items-center ">
      <div class="col-10 col-md-5 col-lg-4 col-xl-3 small-h ps-3 param-docs-row-id-col"><h3><code>--r1_pval_thresh</code></h3></div>
      <div class="col me-auto">
                  <input id="r1_pval_thresh" name="r1_pval_thresh" type="text" value="0.1" class="hicarinput" >
              </div>
      <div class="col-auto d-flex flex-column align-items-end my-1"><div id="--r1_pval_thresh-body" class="param-docs-body small"><p>cutoff pvalue for fragment (R1)</p></div><div class="param_labels_helpbtn d-flex align-items-center">
        <button class="btn btn-sm btn-outline-info ms-2 mb-1 mt-1" data-bs-toggle="collapse" href="#r1_pval_thresh-help" aria-expanded="false">
          <i class="fas fa-question-circle"></i> Help
        </button></div></div>
        <div class="collapse bg-lightgray col-12 param-docs-help-text p-2 mb-2 small d-print-block" id="r1_pval_thresh-help">
          <p>The minimum FDR cutoff to call significant peak for R1 reads. R1 reads will be more noisy compare to R2 reads. Here we need a loose cutoff. And this will be fixed in the following steps.</p></div>
    </div>
    <div class="row param-docs-row border-bottom align-items-baseline mt-5 param-docs-row-group ">
      <div class="col-12 param-docs-row-id-col"><h2>Options related to compartment, TADs calling, and APA</h2></div>
      <div class="col me-auto"></div>
      <div class="col-auto d-flex flex-column align-items-end my-1"></div>
    </div>
    <div class="row param-docs-row border-bottom align-items-center ">
      <div class="col-10 col-md-5 col-lg-4 col-xl-3 small-h ps-3 param-docs-row-id-col"><h3><code>--res_compartments</code></h3></div>
      <div class="col me-auto">
                  <input id="res_compartments" name="res_compartments" type="text" value="100000" class="hicarinput" >
              </div>
      <div class="col-auto d-flex flex-column align-items-end my-1"><div id="--res_compartments-body" class="param-docs-body small"><p>The resolution for compartments calling</p></div></div>
    </div>
    <div class="row param-docs-row border-bottom align-items-center ">
      <div class="col-10 col-md-5 col-lg-4 col-xl-3 small-h ps-3 param-docs-row-id-col"><h3><code>--res_tads</code></h3></div>
      <div class="col me-auto">
                  <input id="res_tads" name="res_tads" type="text" value="10000" class="hicarinput" >
              </div>
      <div class="col-auto d-flex flex-column align-items-end my-1"><div id="--res_tads-body" class="param-docs-body small"><p>The resolution for TADs calling</p></div><div class="param_labels_helpbtn d-flex align-items-center">
        <button class="btn btn-sm btn-outline-info ms-2 mb-1 mt-1" data-bs-toggle="collapse" href="#res_tads-help" aria-expanded="false">
          <i class="fas fa-question-circle"></i> Help
        </button></div></div>
        <div class="collapse bg-lightgray col-12 param-docs-help-text p-2 mb-2 small d-print-block" id="res_tads-help">
          <p>If tad_tool is set to resolution, the windows will be calculated by 3X, 5X, 10X and 15X resolution. If tad_tool is set to hicfindtads, this will determine the step parameter.</p></div>
    </div>
    <div class="row param-docs-row border-bottom align-items-center ">
      <div class="col-10 col-md-5 col-lg-4 col-xl-3 small-h ps-3 param-docs-row-id-col"><h3><code>--tad_tool</code></h3></div>
      <div class="col me-auto">
              <select id="tad_tool" name="tad_tool" class="hicarinput" >
        <option value="cooltools" >cooltools</option><option value="hicexplorer" selected>hicexplorer</option><option value="homer" >homer</option>
              </select>
        </div>
      <div class="col-auto d-flex flex-column align-items-end my-1"><div id="--tad_tool-body" class="param-docs-body small"><p>The caller for TADs, available choices: 'cooltools', 'hicexplorer', 'homer'</p></div><div class="param_labels_helpbtn d-flex align-items-center">
        <button class="btn btn-sm btn-outline-info ms-2 mb-1 mt-1" data-bs-toggle="collapse" href="#tad_tool-help" aria-expanded="false">
          <i class="fas fa-question-circle"></i> Help
        </button></div></div>
        <div class="collapse bg-lightgray col-12 param-docs-help-text p-2 mb-2 small d-print-block" id="tad_tool-help">
          <p>Possible tools are hicfindtads and cooltools (by insulation function).</p></div>
    </div>
    <div class="row param-docs-row border-bottom align-items-center ">
      <div class="col-10 col-md-5 col-lg-4 col-xl-3 small-h ps-3 param-docs-row-id-col"><h3><code>--apa_peak</code></h3></div>
      <div class="col me-auto">
                  <input id="apa_peak" name="apa_peak" type="text" value="Anchor 1D peaks" class="hicarinput" >
              </div>
      <div class="col-auto d-flex flex-column align-items-end my-1"><div id="--apa_peak-body" class="param-docs-body small"><p>The APA peak path</p></div></div>
    </div>
    <div class="row param-docs-row border-bottom align-items-center ">
      <div class="col-10 col-md-5 col-lg-4 col-xl-3 small-h ps-3 param-docs-row-id-col"><h3><code>--apa_tool</code></h3></div>
      <div class="col me-auto">
              <select id="apa_tool" name="apa_tool" class="hicarinput" >
        <option value="cooltools" >cooltools</option><option value="hicexplorer" selected>hicexplorer</option><option value="juicebox" >juicebox</option>
              </select>
        </div>
      <div class="col-auto d-flex flex-column align-items-end my-1"><div id="--apa_tool-body" class="param-docs-body small"><p>APA tool, available choices: 'cooltools', 'hicexplorer', and 'juicebox'</p></div></div>
    </div>
    <div class="row param-docs-row border-bottom align-items-center ">
      <div class="col-10 col-md-5 col-lg-4 col-xl-3 small-h ps-3 param-docs-row-id-col"><h3><code>--apa_format</code></h3></div>
      <div class="col me-auto">
                  <input id="apa_format" name="apa_format" type="text" value="png" class="hicarinput" >
              </div>
      <div class="col-auto d-flex flex-column align-items-end my-1"><div id="--apa_format-body" class="param-docs-body small"><p>The APA output figure format. Currently this parameter will not affect the Juicer output format (always png format).</p></div></div>
    </div>
    <div class="row param-docs-row border-bottom align-items-center ">
      <div class="col-10 col-md-5 col-lg-4 col-xl-3 small-h ps-3 param-docs-row-id-col"><h3><code>--compartments_tool</code></h3></div>
      <div class="col me-auto">
              <select id="compartments_tool" name="compartments_tool" class="hicarinput" >
        <option value="cooltools" selected>cooltools</option><option value="hicexplorer" >hicexplorer</option><option value="homer" >homer</option><option value="juicebox" >juicebox</option>
              </select>
        </div>
      <div class="col-auto d-flex flex-column align-items-end my-1"><div id="--compartments_tool-body" class="param-docs-body small"><p>Call compartment tool, available choices: 'cooltools', 'hicexplorer', 'homer', and 'juicebox'</p></div></div>
    </div>
    <div class="row param-docs-row border-bottom align-items-center ">
      <div class="col-10 col-md-5 col-lg-4 col-xl-3 small-h ps-3 param-docs-row-id-col"><h3><code>--interactions_tool</code></h3></div>
      <div class="col me-auto">
              <select id="interactions_tool" name="interactions_tool" class="hicarinput" >
        <option value="maps" selected>maps</option><option value="hicdcplus" >hicdcplus</option><option value="peakachu" >peakachu</option>
              </select>
        </div>
      <div class="col-auto d-flex flex-column align-items-end my-1"><div id="--interactions_tool-body" class="param-docs-body small"><p>Call loops tool, available choices: 'maps', 'hicdcplus', and 'peakachu'</p></div></div>
    </div>
    <div class="row param-docs-row border-bottom align-items-center ">
      <div class="col-10 col-md-5 col-lg-4 col-xl-3 small-h ps-3 param-docs-row-id-col"><h3><code>--da_tool</code></h3></div>
      <div class="col me-auto">
              <select id="da_tool" name="da_tool" class="hicarinput" >
        <option value="edger" >edger</option><option value="diffhic" selected>diffhic</option><option value="hicexplorer" >hicexplorer</option><option value="setOperation" >setOperation</option>
              </select>
        </div>
      <div class="col-auto d-flex flex-column align-items-end my-1"><div id="--da_tool-body" class="param-docs-body small"><p>Differential analysis tool. Possible options are 'edger', 'diffhic', 'hicexplorer', and 'setOperation'</p></div></div>
    </div>
    <div class="row param-docs-row border-bottom align-items-center ">
      <div class="col-10 col-md-5 col-lg-4 col-xl-3 small-h ps-3 param-docs-row-id-col"><h3><code>--v4c_tool</code></h3></div>
      <div class="col me-auto">
              <select id="v4c_tool" name="v4c_tool" class="hicarinput" >
        <option value="cooltools" >cooltools</option><option value="hicexplorer" selected>hicexplorer</option><option value="trackviewer" >trackviewer</option>
              </select>
        </div>
      <div class="col-auto d-flex flex-column align-items-end my-1"><div id="--v4c_tool-body" class="param-docs-body small"><p>virtual 4c tool, available choices: 'cooltools', 'hicexplorer', and 'trackviewer'</p></div></div>
    </div>
    <div class="row param-docs-row border-bottom align-items-center ">
      <div class="col-10 col-md-5 col-lg-4 col-xl-3 small-h ps-3 param-docs-row-id-col"><h3><code>--tfea_tool</code></h3></div>
      <div class="col me-auto">
              <select id="tfea_tool" name="tfea_tool" class="hicarinput" >
        <option value="atacseqtfea" >atacseqtfea</option><option value="homer" selected>homer</option>
              </select>
        </div>
      <div class="col-auto d-flex flex-column align-items-end my-1"><div id="--tfea_tool-body" class="param-docs-body small"><p>Transcription Factor Enrichment Analysis tool, available choices: 'atacseqtfea', and 'homer'</p></div></div>
    </div>
    <div class="row param-docs-row border-bottom align-items-center ">
      <div class="col-10 col-md-5 col-lg-4 col-xl-3 small-h ps-3 param-docs-row-id-col"><h3><code>--juicer_tools_jar</code></h3></div>
      <div class="col me-auto">
                  <input id="juicer_tools_jar" name="juicer_tools_jar" type="text" value="https://github.com/aidenlab/JuicerTools/releases/download/v3.0.0/juicer_tools.3.0.0.jar" class="hicarinput" >
              </div>
      <div class="col-auto d-flex flex-column align-items-end my-1"><div id="--juicer_tools_jar-body" class="param-docs-body small"><p>juicer_tools jar file url</p></div><div class="param_labels_helpbtn d-flex align-items-center">
        <button class="btn btn-sm btn-outline-info ms-2 mb-1 mt-1" data-bs-toggle="collapse" href="#juicer_tools_jar-help" aria-expanded="false">
          <i class="fas fa-question-circle"></i> Help
        </button></div></div>
        <div class="collapse bg-lightgray col-12 param-docs-help-text p-2 mb-2 small d-print-block" id="juicer_tools_jar-help">
          <p>The juicer_tools is ued for downstream analysis such as apa/arrowhead/hiccups. The juicer_tools can be downloaded from <a href="https://github.com/aidenlab/JuicerTools/releases/">juicer_tools download page</a></p></div>
    </div>
    <div class="row param-docs-row border-bottom align-items-center ">
      <div class="col-10 col-md-5 col-lg-4 col-xl-3 small-h ps-3 param-docs-row-id-col"><h3><code>--juicer_norm_method</code></h3></div>
      <div class="col me-auto">
              <select id="juicer_norm_method" name="juicer_norm_method" class="hicarinput" >
        <option value="VC" >VC</option><option value="VC_SQRT" >VC_SQRT</option><option value="KR" >KR</option><option value="SCALE" selected>SCALE</option><option value="NONE" >NONE</option>
              </select>
        </div>
      <div class="col-auto d-flex flex-column align-items-end my-1"><div id="--juicer_norm_method-body" class="param-docs-body small"><p>Normalization methods</p></div><div class="param_labels_helpbtn d-flex align-items-center">
        <button class="btn btn-sm btn-outline-info ms-2 mb-1 mt-1" data-bs-toggle="collapse" href="#juicer_norm_method-help" aria-expanded="false">
          <i class="fas fa-question-circle"></i> Help
        </button></div></div>
        <div class="collapse bg-lightgray col-12 param-docs-help-text p-2 mb-2 small d-print-block" id="juicer_norm_method-help">
          <p>See more details at <a href="https://github.com/aidenlab/juicer/wiki/AddNorm">addNorm</a>. Avaiable selection are limited by APA and Eigenvector normalization methods.</p></div>
    </div>
    <div class="row param-docs-row border-bottom align-items-center ">
      <div class="col-10 col-md-5 col-lg-4 col-xl-3 small-h ps-3 param-docs-row-id-col"><h3><code>--peak_interactions_threshold</code></h3></div>
      <div class="col me-auto">
                  <input id="peak_interactions_threshold" name="peak_interactions_threshold" type="text" value="2" class="hicarinput" >
              </div>
      <div class="col-auto d-flex flex-column align-items-end my-1"><div id="--peak_interactions_threshold-body" class="param-docs-body small"><p>parameter for HiCExplorer ChicSignificantinterations</p></div></div>
    </div>
    <div class="row param-docs-row border-bottom align-items-center param-docs-hidden collapse d-print-block ">
      <div class="col-10 col-md-5 col-lg-4 col-xl-3 small-h ps-3 param-docs-row-id-col"><h3><code>--maps_3d_ext</code></h3></div>
      <div class="col me-auto"></div>
      <div class="col-auto d-flex flex-column align-items-end my-1"><div id="--maps_3d_ext-body" class="param-docs-body small"><p>extension file name for MAPS</p></div><div class="param_labels_helpbtn d-flex align-items-center"><span class="badge bg-light text-dark small ms-2">hidden</span></div></div>
    </div>
    <div class="row param-docs-row border-bottom align-items-center param-docs-hidden collapse d-print-block ">
      <div class="col-10 col-md-5 col-lg-4 col-xl-3 small-h ps-3 param-docs-row-id-col"><h3><code>--long_bedpe_postfix</code></h3></div>
      <div class="col me-auto"></div>
      <div class="col-auto d-flex flex-column align-items-end my-1"><div id="--long_bedpe_postfix-body" class="param-docs-body small"><p>extension file name for long pairs</p></div><div class="param_labels_helpbtn d-flex align-items-center"><span class="badge bg-light text-dark small ms-2">hidden</span></div></div>
    </div>
    <div class="row param-docs-row border-bottom align-items-center param-docs-hidden collapse d-print-block ">
      <div class="col-10 col-md-5 col-lg-4 col-xl-3 small-h ps-3 param-docs-row-id-col"><h3><code>--short_bed_postfix</code></h3></div>
      <div class="col me-auto"></div>
      <div class="col-auto d-flex flex-column align-items-end my-1"><div id="--short_bed_postfix-body" class="param-docs-body small"><p>extension file name for short pairs</p></div><div class="param_labels_helpbtn d-flex align-items-center"><span class="badge bg-light text-dark small ms-2">hidden</span></div></div>
    </div>
    <div class="row param-docs-row border-bottom align-items-baseline mt-5 param-docs-row-group ">
      <div class="col-12 param-docs-row-id-col"><h2>Options related to tracks, HiCTools, and circos</h2></div>
      <div class="col me-auto"></div>
      <div class="col-auto d-flex flex-column align-items-end my-1"></div>
    </div>
    <div class="row param-docs-row border-bottom align-items-center ">
      <div class="col-10 col-md-5 col-lg-4 col-xl-3 small-h ps-3 param-docs-row-id-col"><h3><code>--v4c_max_events</code></h3></div>
      <div class="col me-auto">
                  <input id="v4c_max_events" name="v4c_max_events" type="text" value="25" class="hicarinput" >
              </div>
      <div class="col-auto d-flex flex-column align-items-end my-1"><div id="--v4c_max_events-body" class="param-docs-body small"><p>max events to plot for virtual 4c</p></div></div>
    </div>
    <div class="row param-docs-row border-bottom align-items-center ">
      <div class="col-10 col-md-5 col-lg-4 col-xl-3 small-h ps-3 param-docs-row-id-col"><h3><code>--hic_tools_jar</code></h3></div>
      <div class="col me-auto">
                  <input id="hic_tools_jar" name="hic_tools_jar" type="text" value="https://github.com/aidenlab/HiCTools/releases/download/v3.30.00/hic_tools.3.30.00.jar" class="hicarinput" >
              </div>
      <div class="col-auto d-flex flex-column align-items-end my-1"><div id="--hic_tools_jar-body" class="param-docs-body small"><p>The HiCTools path</p></div><div class="param_labels_helpbtn d-flex align-items-center">
        <button class="btn btn-sm btn-outline-info ms-2 mb-1 mt-1" data-bs-toggle="collapse" href="#hic_tools_jar-help" aria-expanded="false">
          <i class="fas fa-question-circle"></i> Help
        </button></div></div>
        <div class="collapse bg-lightgray col-12 param-docs-help-text p-2 mb-2 small d-print-block" id="hic_tools_jar-help">
          <p>This tool is used for pre/addNorm/eigenvector/pearsons/sum. Pre is the tool to create .hic file; eigenvector is used for compartment caller. The hic_tools can be downloaded from <a href="https://github.com/aidenlab/HiCTools/releases">HiCTools download page</a></p></div>
    </div>
    <div class="row param-docs-row border-bottom align-items-baseline mt-5 param-docs-row-group param-docs-hidden collapse d-print-block ">
      <div class="col-12 param-docs-row-id-col"><h2>Other options not expose</h2></div>
      <div class="col me-auto"></div>
      <div class="col-auto d-flex flex-column align-items-end my-1"><div class="param_labels_helpbtn d-flex align-items-center"><span class="badge bg-light text-dark small ms-2">hidden</span></div></div>
    </div>
    <div class="row param-docs-row border-bottom align-items-center param-docs-hidden collapse d-print-block ">
      <div class="col-10 col-md-5 col-lg-4 col-xl-3 small-h ps-3 param-docs-row-id-col"><h3><code>--skip_peak_qc</code></h3></div>
      <div class="col me-auto"></div>
      <div class="col-auto d-flex flex-column align-items-end my-1"><div id="--skip_peak_qc-body" class="param-docs-body small"><p>skip peak QC or not</p></div><div class="param_labels_helpbtn d-flex align-items-center"><span class="badge bg-light text-dark small ms-2">hidden</span></div></div>
    </div>
    <div class="row param-docs-row border-bottom align-items-center param-docs-hidden collapse d-print-block ">
      <div class="col-10 col-md-5 col-lg-4 col-xl-3 small-h ps-3 param-docs-row-id-col"><h3><code>--skip_plot_profile</code></h3></div>
      <div class="col me-auto"></div>
      <div class="col-auto d-flex flex-column align-items-end my-1"><div id="--skip_plot_profile-body" class="param-docs-body small"><p>skip plot profile or not</p></div><div class="param_labels_helpbtn d-flex align-items-center"><span class="badge bg-light text-dark small ms-2">hidden</span></div></div>
    </div>
    <div class="row param-docs-row border-bottom align-items-center param-docs-hidden collapse d-print-block ">
      <div class="col-10 col-md-5 col-lg-4 col-xl-3 small-h ps-3 param-docs-row-id-col"><h3><code>--skip_igv</code></h3></div>
      <div class="col me-auto"></div>
      <div class="col-auto d-flex flex-column align-items-end my-1"><div id="--skip_igv-body" class="param-docs-body small"><p>skip creat IGV files or not</p></div><div class="param_labels_helpbtn d-flex align-items-center"><span class="badge bg-light text-dark small ms-2">hidden</span></div></div>
    </div>
    <div class="row param-docs-row border-bottom align-items-center param-docs-hidden collapse d-print-block ">
      <div class="col-10 col-md-5 col-lg-4 col-xl-3 small-h ps-3 param-docs-row-id-col"><h3><code>--skip_trackhub</code></h3></div>
      <div class="col me-auto"></div>
      <div class="col-auto d-flex flex-column align-items-end my-1"><div id="--skip_trackhub-body" class="param-docs-body small"><p>skip create trackhub files or not</p></div><div class="param_labels_helpbtn d-flex align-items-center"><span class="badge bg-light text-dark small ms-2">hidden</span></div></div>
    </div>
    <div class="row param-docs-row border-bottom align-items-center param-docs-hidden collapse d-print-block ">
      <div class="col-10 col-md-5 col-lg-4 col-xl-3 small-h ps-3 param-docs-row-id-col"><h3><code>--enrichment_fdr</code></h3></div>
      <div class="col me-auto"></div>
      <div class="col-auto d-flex flex-column align-items-end my-1"><div id="--enrichment_fdr-body" class="param-docs-body small"><p>cutoff value for false discovery rate of enrichment analysis</p></div><div class="param_labels_helpbtn d-flex align-items-center"><span class="badge bg-light text-dark small ms-2">hidden</span></div></div>
    </div>
    <div class="row param-docs-row border-bottom align-items-baseline mt-5 param-docs-row-group param-docs-hidden collapse d-print-block ">
      <div class="col-12 param-docs-row-id-col"><h2>Institutional config options</h2></div>
      <div class="col me-auto"></div>
      <div class="col-auto d-flex flex-column align-items-end my-1"><div class="param_labels_helpbtn d-flex align-items-center">
        <button class="btn btn-sm btn-outline-info ms-2 mb-1 mt-1" data-bs-toggle="collapse" href="#Institutional config options-help" aria-expanded="false">
          <i class="fas fa-question-circle"></i> Help
        </button><span class="badge bg-light text-dark small ms-2">hidden</span></div></div>
        <div class="collapse bg-lightgray col-12 param-docs-help-text p-2 mb-2 small d-print-block" id="Institutional config options-help">
          <p>The centralised nf-core configuration profiles use a handful of pipeline parameters to describe themselves. This information is then printed to the Nextflow log when you run a pipeline. You should not need to change these values when you run a pipeline.</p></div>
    </div>
    <div class="row param-docs-row border-bottom align-items-center param-docs-hidden collapse d-print-block ">
      <div class="col-10 col-md-5 col-lg-4 col-xl-3 small-h ps-3 param-docs-row-id-col"><h3><code>--custom_config_version</code></h3></div>
      <div class="col me-auto"></div>
      <div class="col-auto d-flex flex-column align-items-end my-1"><div id="--custom_config_version-body" class="param-docs-body small"><p>Git commit id for Institutional configs.</p></div><div class="param_labels_helpbtn d-flex align-items-center"><span class="badge bg-light text-dark small ms-2">hidden</span></div></div>
    </div>
    <div class="row param-docs-row border-bottom align-items-center param-docs-hidden collapse d-print-block ">
      <div class="col-10 col-md-5 col-lg-4 col-xl-3 small-h ps-3 param-docs-row-id-col"><h3><code>--custom_config_base</code></h3></div>
      <div class="col me-auto"></div>
      <div class="col-auto d-flex flex-column align-items-end my-1"><div id="--custom_config_base-body" class="param-docs-body small"><p>Base directory for Institutional configs.</p></div><div class="param_labels_helpbtn d-flex align-items-center">
        <button class="btn btn-sm btn-outline-info ms-2 mb-1 mt-1" data-bs-toggle="collapse" href="#custom_config_base-help" aria-expanded="false">
          <i class="fas fa-question-circle"></i> Help
        </button><span class="badge bg-light text-dark small ms-2">hidden</span></div></div>
        <div class="collapse bg-lightgray col-12 param-docs-help-text p-2 mb-2 small d-print-block" id="custom_config_base-help">
          <p>If you're running offline, Nextflow will not be able to fetch the institutional config files from the internet. If you don't need them, then this is not a problem. If you do need them, you should download the files from the repo and tell Nextflow where to find them with this parameter.</p></div>
    </div>
    <div class="row param-docs-row border-bottom align-items-center param-docs-hidden collapse d-print-block ">
      <div class="col-10 col-md-5 col-lg-4 col-xl-3 small-h ps-3 param-docs-row-id-col"><h3><code>--config_profile_name</code></h3></div>
      <div class="col me-auto"></div>
      <div class="col-auto d-flex flex-column align-items-end my-1"><div id="--config_profile_name-body" class="param-docs-body small"><p>Institutional config name.</p></div><div class="param_labels_helpbtn d-flex align-items-center"><span class="badge bg-light text-dark small ms-2">hidden</span></div></div>
    </div>
    <div class="row param-docs-row border-bottom align-items-center param-docs-hidden collapse d-print-block ">
      <div class="col-10 col-md-5 col-lg-4 col-xl-3 small-h ps-3 param-docs-row-id-col"><h3><code>--config_profile_description</code></h3></div>
      <div class="col me-auto"></div>
      <div class="col-auto d-flex flex-column align-items-end my-1"><div id="--config_profile_description-body" class="param-docs-body small"><p>Institutional config description.</p></div><div class="param_labels_helpbtn d-flex align-items-center"><span class="badge bg-light text-dark small ms-2">hidden</span></div></div>
    </div>
    <div class="row param-docs-row border-bottom align-items-center param-docs-hidden collapse d-print-block ">
      <div class="col-10 col-md-5 col-lg-4 col-xl-3 small-h ps-3 param-docs-row-id-col"><h3><code>--config_profile_contact</code></h3></div>
      <div class="col me-auto"></div>
      <div class="col-auto d-flex flex-column align-items-end my-1"><div id="--config_profile_contact-body" class="param-docs-body small"><p>Institutional config contact information.</p></div><div class="param_labels_helpbtn d-flex align-items-center"><span class="badge bg-light text-dark small ms-2">hidden</span></div></div>
    </div>
    <div class="row param-docs-row border-bottom align-items-center param-docs-hidden collapse d-print-block ">
      <div class="col-10 col-md-5 col-lg-4 col-xl-3 small-h ps-3 param-docs-row-id-col"><h3><code>--config_profile_url</code></h3></div>
      <div class="col me-auto"></div>
      <div class="col-auto d-flex flex-column align-items-end my-1"><div id="--config_profile_url-body" class="param-docs-body small"><p>Institutional config URL link.</p></div><div class="param_labels_helpbtn d-flex align-items-center"><span class="badge bg-light text-dark small ms-2">hidden</span></div></div>
    </div>
    <div class="row param-docs-row border-bottom align-items-baseline mt-5 param-docs-row-group param-docs-hidden collapse d-print-block ">
      <div class="col-12 param-docs-row-id-col"><h2>Max job request options</h2></div>
      <div class="col me-auto"></div>
      <div class="col-auto d-flex flex-column align-items-end my-1"><div class="param_labels_helpbtn d-flex align-items-center">
        <button class="btn btn-sm btn-outline-info ms-2 mb-1 mt-1" data-bs-toggle="collapse" href="#Max job request options-help" aria-expanded="false">
          <i class="fas fa-question-circle"></i> Help
        </button><span class="badge bg-light text-dark small ms-2">hidden</span></div></div>
        <div class="collapse bg-lightgray col-12 param-docs-help-text p-2 mb-2 small d-print-block" id="Max job request options-help">
          <p>If you are running on a smaller system, a pipeline step requesting more resources than are available may cause the Nextflow to stop the run with an error. These options allow you to cap the maximum resources requested by any single job so that the pipeline will run on your system.</p>
<p>Note that you can not <em>increase</em> the resources requested by any job using these options. For that you will need your own configuration file. See <a href="https://nf-co.re/usage/configuration">the nf-core website</a> for details.</p></div>
    </div>
    <div class="row param-docs-row border-bottom align-items-center param-docs-hidden collapse d-print-block ">
      <div class="col-10 col-md-5 col-lg-4 col-xl-3 small-h ps-3 param-docs-row-id-col"><h3><code>--max_cpus</code></h3></div>
      <div class="col me-auto"></div>
      <div class="col-auto d-flex flex-column align-items-end my-1"><div id="--max_cpus-body" class="param-docs-body small"><p>Maximum number of CPUs that can be requested for any single job.</p></div><div class="param_labels_helpbtn d-flex align-items-center">
        <button class="btn btn-sm btn-outline-info ms-2 mb-1 mt-1" data-bs-toggle="collapse" href="#max_cpus-help" aria-expanded="false">
          <i class="fas fa-question-circle"></i> Help
        </button><span class="badge bg-light text-dark small ms-2">hidden</span></div></div>
        <div class="collapse bg-lightgray col-12 param-docs-help-text p-2 mb-2 small d-print-block" id="max_cpus-help">
          <p>Use to set an upper-limit for the CPU requirement for each process. Should be an integer e.g. <code><span class="text-nowrap">--max_cpus</span> 1</code></p></div>
    </div>
    <div class="row param-docs-row border-bottom align-items-center param-docs-hidden collapse d-print-block ">
      <div class="col-10 col-md-5 col-lg-4 col-xl-3 small-h ps-3 param-docs-row-id-col"><h3><code>--max_memory</code></h3></div>
      <div class="col me-auto"></div>
      <div class="col-auto d-flex flex-column align-items-end my-1"><div id="--max_memory-body" class="param-docs-body small"><p>Maximum amount of memory that can be requested for any single job.</p></div><div class="param_labels_helpbtn d-flex align-items-center">
        <button class="btn btn-sm btn-outline-info ms-2 mb-1 mt-1" data-bs-toggle="collapse" href="#max_memory-help" aria-expanded="false">
          <i class="fas fa-question-circle"></i> Help
        </button><span class="badge bg-light text-dark small ms-2">hidden</span></div></div>
        <div class="collapse bg-lightgray col-12 param-docs-help-text p-2 mb-2 small d-print-block" id="max_memory-help">
          <p>Use to set an upper-limit for the memory requirement for each process. Should be a string in the format integer-unit e.g. <code><span class="text-nowrap">--max_memory</span> '8.GB'</code></p></div>
    </div>
    <div class="row param-docs-row border-bottom align-items-center param-docs-hidden collapse d-print-block ">
      <div class="col-10 col-md-5 col-lg-4 col-xl-3 small-h ps-3 param-docs-row-id-col"><h3><code>--max_time</code></h3></div>
      <div class="col me-auto"></div>
      <div class="col-auto d-flex flex-column align-items-end my-1"><div id="--max_time-body" class="param-docs-body small"><p>Maximum amount of time that can be requested for any single job.</p></div><div class="param_labels_helpbtn d-flex align-items-center">
        <button class="btn btn-sm btn-outline-info ms-2 mb-1 mt-1" data-bs-toggle="collapse" href="#max_time-help" aria-expanded="false">
          <i class="fas fa-question-circle"></i> Help
        </button><span class="badge bg-light text-dark small ms-2">hidden</span></div></div>
        <div class="collapse bg-lightgray col-12 param-docs-help-text p-2 mb-2 small d-print-block" id="max_time-help">
          <p>Use to set an upper-limit for the time requirement for each process. Should be a string in the format integer-unit e.g. <code><span class="text-nowrap">--max_time</span> '2.h'</code></p></div>
    </div>
    <div class="row param-docs-row border-bottom align-items-baseline mt-5 param-docs-row-group ">
      <div class="col-12 param-docs-row-id-col"><h2>Generic options</h2></div>
      <div class="col me-auto"></div>
      <div class="col-auto d-flex flex-column align-items-end my-1"></div>
        <div class="collapse bg-lightgray col-12 param-docs-help-text p-2 mb-2 small d-print-block" id="Generic options-help">
          <p>These options are common to all nf-core pipelines and allow you to customise some of the core preferences for how the pipeline runs.</p>
<p>Typically these options would be set in a Nextflow config file loaded for all pipeline runs, such as <code>~/.nextflow/config</code>.</p></div>
    </div>
    <div class="row param-docs-row border-bottom align-items-center param-docs-hidden collapse d-print-block ">
      <div class="col-10 col-md-5 col-lg-4 col-xl-3 small-h ps-3 param-docs-row-id-col"><h3><code>--help</code></h3></div>
      <div class="col me-auto"></div>
      <div class="col-auto d-flex flex-column align-items-end my-1"><div id="--help-body" class="param-docs-body small"><p>Display help text.</p></div><div class="param_labels_helpbtn d-flex align-items-center"><span class="badge bg-light text-dark small ms-2">hidden</span></div></div>
    </div>
    <div class="row param-docs-row border-bottom align-items-center param-docs-hidden collapse d-print-block ">
      <div class="col-10 col-md-5 col-lg-4 col-xl-3 small-h ps-3 param-docs-row-id-col"><h3><code>--version</code></h3></div>
      <div class="col me-auto"></div>
      <div class="col-auto d-flex flex-column align-items-end my-1"><div id="--version-body" class="param-docs-body small"><p>Display version and exit.</p></div><div class="param_labels_helpbtn d-flex align-items-center"><span class="badge bg-light text-dark small ms-2">hidden</span></div></div>
    </div>
    <div class="row param-docs-row border-bottom align-items-center param-docs-hidden collapse d-print-block ">
      <div class="col-10 col-md-5 col-lg-4 col-xl-3 small-h ps-3 param-docs-row-id-col"><h3><code>--publish_dir_mode</code></h3></div>
      <div class="col me-auto"></div>
      <div class="col-auto d-flex flex-column align-items-end my-1"><div id="--publish_dir_mode-body" class="param-docs-body small"><p>Method used to save pipeline results to output directory.</p></div><div class="param_labels_helpbtn d-flex align-items-center">
        <button class="btn btn-sm btn-outline-info ms-2 mb-1 mt-1" data-bs-toggle="collapse" href="#publish_dir_mode-help" aria-expanded="false">
          <i class="fas fa-question-circle"></i> Help
        </button><span class="badge bg-light text-dark small ms-2">hidden</span></div></div>
        <div class="collapse bg-lightgray col-12 param-docs-help-text p-2 mb-2 small d-print-block" id="publish_dir_mode-help">
          <p>The Nextflow <code>publishDir</code> option specifies which intermediate files should be saved to the output directory. This option tells the pipeline what method should be used to move these files. See <a href="https://www.nextflow.io/docs/latest/process.html#publishdir">Nextflow docs</a> for details.</p></div>
    </div>
    <div class="row param-docs-row border-bottom align-items-center param-docs-hidden collapse d-print-block ">
      <div class="col-10 col-md-5 col-lg-4 col-xl-3 small-h ps-3 param-docs-row-id-col"><h3><code>--email_on_fail</code></h3></div>
      <div class="col me-auto"></div>
      <div class="col-auto d-flex flex-column align-items-end my-1"><div id="--email_on_fail-body" class="param-docs-body small"><p>Email address for completion summary, only when pipeline fails.</p></div><div class="param_labels_helpbtn d-flex align-items-center">
        <button class="btn btn-sm btn-outline-info ms-2 mb-1 mt-1" data-bs-toggle="collapse" href="#email_on_fail-help" aria-expanded="false">
          <i class="fas fa-question-circle"></i> Help
        </button><span class="badge bg-light text-dark small ms-2">hidden</span></div></div>
        <div class="collapse bg-lightgray col-12 param-docs-help-text p-2 mb-2 small d-print-block" id="email_on_fail-help">
          <p>An email address to send a summary email to when the pipeline is completed - ONLY sent if the pipeline does not exit successfully.</p></div>
    </div>
    <div class="row param-docs-row border-bottom align-items-center param-docs-hidden collapse d-print-block ">
      <div class="col-10 col-md-5 col-lg-4 col-xl-3 small-h ps-3 param-docs-row-id-col"><h3><code>--plaintext_email</code></h3></div>
      <div class="col me-auto"></div>
      <div class="col-auto d-flex flex-column align-items-end my-1"><div id="--plaintext_email-body" class="param-docs-body small"><p>Send plain-text email instead of HTML.</p></div><div class="param_labels_helpbtn d-flex align-items-center"><span class="badge bg-light text-dark small ms-2">hidden</span></div></div>
    </div>
    <div class="row param-docs-row border-bottom align-items-center param-docs-hidden collapse d-print-block ">
      <div class="col-10 col-md-5 col-lg-4 col-xl-3 small-h ps-3 param-docs-row-id-col"><h3><code>--max_multiqc_email_size</code></h3></div>
      <div class="col me-auto"></div>
      <div class="col-auto d-flex flex-column align-items-end my-1"><div id="--max_multiqc_email_size-body" class="param-docs-body small"><p>File size limit when attaching MultiQC reports to summary emails.</p></div><div class="param_labels_helpbtn d-flex align-items-center"><span class="badge bg-light text-dark small ms-2">hidden</span></div></div>
    </div>
    <div class="row param-docs-row border-bottom align-items-center param-docs-hidden collapse d-print-block ">
      <div class="col-10 col-md-5 col-lg-4 col-xl-3 small-h ps-3 param-docs-row-id-col"><h3><code>--monochrome_logs</code></h3></div>
      <div class="col me-auto"></div>
      <div class="col-auto d-flex flex-column align-items-end my-1"><div id="--monochrome_logs-body" class="param-docs-body small"><p>Do not use coloured log outputs.</p></div><div class="param_labels_helpbtn d-flex align-items-center"><span class="badge bg-light text-dark small ms-2">hidden</span></div></div>
    </div>
    <div class="row param-docs-row border-bottom align-items-center param-docs-hidden collapse d-print-block ">
      <div class="col-10 col-md-5 col-lg-4 col-xl-3 small-h ps-3 param-docs-row-id-col"><h3><code>--hook_url</code></h3></div>
      <div class="col me-auto"></div>
      <div class="col-auto d-flex flex-column align-items-end my-1"><div id="--hook_url-body" class="param-docs-body small"><p>Incoming hook URL for messaging service</p></div><div class="param_labels_helpbtn d-flex align-items-center">
        <button class="btn btn-sm btn-outline-info ms-2 mb-1 mt-1" data-bs-toggle="collapse" href="#hook_url-help" aria-expanded="false">
          <i class="fas fa-question-circle"></i> Help
        </button><span class="badge bg-light text-dark small ms-2">hidden</span></div></div>
        <div class="collapse bg-lightgray col-12 param-docs-help-text p-2 mb-2 small d-print-block" id="hook_url-help">
          <p>Incoming hook URL for messaging service. Currently, MS Teams and Slack are supported.</p></div>
    </div>
    <div class="row param-docs-row border-bottom align-items-center param-docs-hidden collapse d-print-block ">
      <div class="col-10 col-md-5 col-lg-4 col-xl-3 small-h ps-3 param-docs-row-id-col"><h3><code>--multiqc_config</code></h3></div>
      <div class="col me-auto"></div>
      <div class="col-auto d-flex flex-column align-items-end my-1"><div id="--multiqc_config-body" class="param-docs-body small"><p>Custom config file to supply to MultiQC.</p></div><div class="param_labels_helpbtn d-flex align-items-center"><span class="badge bg-light text-dark small ms-2">hidden</span></div></div>
    </div>
    <div class="row param-docs-row border-bottom align-items-center param-docs-hidden collapse d-print-block ">
      <div class="col-10 col-md-5 col-lg-4 col-xl-3 small-h ps-3 param-docs-row-id-col"><h3><code>--multiqc_logo</code></h3></div>
      <div class="col me-auto"></div>
      <div class="col-auto d-flex flex-column align-items-end my-1"><div id="--multiqc_logo-body" class="param-docs-body small"><p>Custom logo file to supply to MultiQC. File name must also be set in the MultiQC config file</p></div><div class="param_labels_helpbtn d-flex align-items-center"><span class="badge bg-light text-dark small ms-2">hidden</span></div></div>
    </div>
    <div class="row param-docs-row border-bottom align-items-center ">
      <div class="col-10 col-md-5 col-lg-4 col-xl-3 small-h ps-3 param-docs-row-id-col"><h3><code>--multiqc_methods_description</code></h3></div>
      <div class="col me-auto">
                  <input id="multiqc_methods_description" name="multiqc_methods_description" type="text" value="" class="hicarinput" >
              </div>
      <div class="col-auto d-flex flex-column align-items-end my-1"><div id="--multiqc_methods_description-body" class="param-docs-body small"><p>Custom MultiQC yaml file containing HTML including a methods description.</p></div></div>
    </div>
    <div class="row param-docs-row border-bottom align-items-center param-docs-hidden collapse d-print-block ">
      <div class="col-10 col-md-5 col-lg-4 col-xl-3 small-h ps-3 param-docs-row-id-col"><h3><code>--validate_params</code></h3></div>
      <div class="col me-auto"></div>
      <div class="col-auto d-flex flex-column align-items-end my-1"><div id="--validate_params-body" class="param-docs-body small"><p>Boolean whether to validate parameters against the schema at runtime</p></div><div class="param_labels_helpbtn d-flex align-items-center"><span class="badge bg-light text-dark small ms-2">hidden</span></div></div>
    </div>
    <div class="row param-docs-row border-bottom align-items-center param-docs-hidden collapse d-print-block ">
      <div class="col-10 col-md-5 col-lg-4 col-xl-3 small-h ps-3 param-docs-row-id-col"><h3><code>--validationShowHiddenParams</code></h3></div>
      <div class="col me-auto"></div>
      <div class="col-auto d-flex flex-column align-items-end my-1"><div id="--validationShowHiddenParams-body" class="param-docs-body small"><p>Show all params when using <code><span class="text-nowrap">--help</code></p></span></div><div class="param_labels_helpbtn d-flex align-items-center">
        <button class="btn btn-sm btn-outline-info ms-2 mb-1 mt-1" data-bs-toggle="collapse" href="#validationShowHiddenParams-help" aria-expanded="false">
          <i class="fas fa-question-circle"></i> Help
        </button><span class="badge bg-light text-dark small ms-2">hidden</span></div></div>
        <div class="collapse bg-lightgray col-12 param-docs-help-text p-2 mb-2 small d-print-block" id="validationShowHiddenParams-help">
          <p>By default, parameters set as <em>hidden</em> in the schema are not shown on the command line when a user runs with <code><span class="text-nowrap">--help</code>.</span> Specifying this option will tell the pipeline to show all parameters.</p></div>
    </div>
    <div class="row param-docs-row border-bottom align-items-center param-docs-hidden collapse d-print-block ">
      <div class="col-10 col-md-5 col-lg-4 col-xl-3 small-h ps-3 param-docs-row-id-col"><h3><code>--validationFailUnrecognisedParams</code></h3></div>
      <div class="col me-auto"></div>
      <div class="col-auto d-flex flex-column align-items-end my-1"><div id="--validationFailUnrecognisedParams-body" class="param-docs-body small"><p>Validation of parameters fails when an unrecognised parameter is found.</p></div><div class="param_labels_helpbtn d-flex align-items-center">
        <button class="btn btn-sm btn-outline-info ms-2 mb-1 mt-1" data-bs-toggle="collapse" href="#validationFailUnrecognisedParams-help" aria-expanded="false">
          <i class="fas fa-question-circle"></i> Help
        </button><span class="badge bg-light text-dark small ms-2">hidden</span></div></div>
        <div class="collapse bg-lightgray col-12 param-docs-help-text p-2 mb-2 small d-print-block" id="validationFailUnrecognisedParams-help">
          <p>By default, when an unrecognised parameter is found, it returns a warinig.</p></div>
    </div>
    <div class="row param-docs-row border-bottom align-items-center param-docs-hidden collapse d-print-block ">
      <div class="col-10 col-md-5 col-lg-4 col-xl-3 small-h ps-3 param-docs-row-id-col"><h3><code>--validationLenientMode</code></h3></div>
      <div class="col me-auto"></div>
      <div class="col-auto d-flex flex-column align-items-end my-1"><div id="--validationLenientMode-body" class="param-docs-body small"><p>Validation of parameters in lenient more.</p></div><div class="param_labels_helpbtn d-flex align-items-center">
        <button class="btn btn-sm btn-outline-info ms-2 mb-1 mt-1" data-bs-toggle="collapse" href="#validationLenientMode-help" aria-expanded="false">
          <i class="fas fa-question-circle"></i> Help
        </button><span class="badge bg-light text-dark small ms-2">hidden</span></div></div>
        <div class="collapse bg-lightgray col-12 param-docs-help-text p-2 mb-2 small d-print-block" id="validationLenientMode-help">
          <p>Allows string values that are parseable as numbers or booleans. For further information see <a href="https://github.com/everit-org/json-schema#lenient-mode">JSONSchema docs</a>.</p></div>
    </div>
    <div class="row param-docs-row border-bottom align-items-baseline mt-5 param-docs-row-group ">
      <div class="col-12 param-docs-row-id-col"><h2>Pipeline controler</h2></div>
      <div class="col me-auto"></div>
      <div class="col-auto d-flex flex-column align-items-end my-1"></div>
        <div class="collapse bg-lightgray col-12 param-docs-help-text p-2 mb-2 small d-print-block" id="Pipeline controler-help">
          <p>These options allow you to customise some of the pipeline steps for how the pipeline runs.</p></div>
    </div>
    <div class="row param-docs-row border-bottom align-items-center ">
      <div class="col-10 col-md-5 col-lg-4 col-xl-3 small-h ps-3 param-docs-row-id-col"><h3><code>--skip_cutadapt</code></h3></div>
      <div class="col me-auto">
                  <input id="skip_cutadapt" name="skip_cutadapt" type="checkbox" class="hicarinput" >
              </div>
      <div class="col-auto d-flex flex-column align-items-end my-1"><div id="--skip_cutadapt-body" class="param-docs-body small"><p>skip trim 5'end</p></div></div>
    </div>
    <div class="row param-docs-row border-bottom align-items-center ">
      <div class="col-10 col-md-5 col-lg-4 col-xl-3 small-h ps-3 param-docs-row-id-col"><h3><code>--resample_pairs</code></h3></div>
      <div class="col me-auto">
                  <input id="resample_pairs" name="resample_pairs" type="checkbox" class="hicarinput" >
              </div>
      <div class="col-auto d-flex flex-column align-items-end my-1"><div id="--resample_pairs-body" class="param-docs-body small"><p>resample the pairs by pairtools</p></div></div>
    </div>
    <div class="row param-docs-row border-bottom align-items-center ">
      <div class="col-10 col-md-5 col-lg-4 col-xl-3 small-h ps-3 param-docs-row-id-col"><h3><code>--skip_fastqc</code></h3></div>
      <div class="col me-auto">
                  <input id="skip_fastqc" name="skip_fastqc" type="checkbox" class="hicarinput" >
              </div>
      <div class="col-auto d-flex flex-column align-items-end my-1"><div id="--skip_fastqc-body" class="param-docs-body small"><p>skip fastqc or not</p></div></div>
    </div>
    <div class="row param-docs-row border-bottom align-items-center ">
      <div class="col-10 col-md-5 col-lg-4 col-xl-3 small-h ps-3 param-docs-row-id-col"><h3><code>--skip_peak_annotation</code></h3></div>
      <div class="col me-auto">
                  <input id="skip_peak_annotation" name="skip_peak_annotation" type="checkbox" class="hicarinput" >
              </div>
      <div class="col-auto d-flex flex-column align-items-end my-1"><div id="--skip_peak_annotation-body" class="param-docs-body small"><p>skip peak annotation or not</p></div></div>
    </div>
    <div class="row param-docs-row border-bottom align-items-center ">
      <div class="col-10 col-md-5 col-lg-4 col-xl-3 small-h ps-3 param-docs-row-id-col"><h3><code>--skip_diff_analysis</code></h3></div>
      <div class="col me-auto">
                  <input id="skip_diff_analysis" name="skip_diff_analysis" type="checkbox" class="hicarinput" >
              </div>
      <div class="col-auto d-flex flex-column align-items-end my-1"><div id="--skip_diff_analysis-body" class="param-docs-body small"><p>skip differential analysis or not</p></div></div>
    </div>
    <div class="row param-docs-row border-bottom align-items-center ">
      <div class="col-10 col-md-5 col-lg-4 col-xl-3 small-h ps-3 param-docs-row-id-col"><h3><code>--skip_multiqc</code></h3></div>
      <div class="col me-auto">
                  <input id="skip_multiqc" name="skip_multiqc" type="checkbox" class="hicarinput" >
              </div>
      <div class="col-auto d-flex flex-column align-items-end my-1"><div id="--skip_multiqc-body" class="param-docs-body small"><p>skip multiqc or not</p></div></div>
    </div>
    <div class="row param-docs-row border-bottom align-items-center ">
      <div class="col-10 col-md-5 col-lg-4 col-xl-3 small-h ps-3 param-docs-row-id-col"><h3><code>--do_apa</code></h3></div>
      <div class="col me-auto">
                  <input id="do_apa" name="do_apa" type="checkbox" class="hicarinput" >
              </div>
      <div class="col-auto d-flex flex-column align-items-end my-1"><div id="--do_apa-body" class="param-docs-body small"><p>run APA or not</p></div></div>
    </div>
    <div class="row param-docs-row border-bottom align-items-center ">
      <div class="col-10 col-md-5 col-lg-4 col-xl-3 small-h ps-3 param-docs-row-id-col"><h3><code>--skip_compartments</code></h3></div>
      <div class="col me-auto">
                  <input id="skip_compartments" name="skip_compartments" type="checkbox" class="hicarinput" >
              </div>
      <div class="col-auto d-flex flex-column align-items-end my-1"><div id="--skip_compartments-body" class="param-docs-body small"><p>skip call compartment or not</p></div></div>
    </div>
    <div class="row param-docs-row border-bottom align-items-center ">
      <div class="col-10 col-md-5 col-lg-4 col-xl-3 small-h ps-3 param-docs-row-id-col"><h3><code>--skip_tads</code></h3></div>
      <div class="col me-auto">
                  <input id="skip_tads" name="skip_tads" type="checkbox" class="hicarinput" >
              </div>
      <div class="col-auto d-flex flex-column align-items-end my-1"><div id="--skip_tads-body" class="param-docs-body small"><p>skip call TADs or not</p></div></div>
    </div>
    <div class="row param-docs-row border-bottom align-items-center ">
      <div class="col-10 col-md-5 col-lg-4 col-xl-3 small-h ps-3 param-docs-row-id-col"><h3><code>--skip_interactions</code></h3></div>
      <div class="col me-auto">
                  <input id="skip_interactions" name="skip_interactions" type="checkbox" class="hicarinput" >
              </div>
      <div class="col-auto d-flex flex-column align-items-end my-1"><div id="--skip_interactions-body" class="param-docs-body small"><p>skip call interactions/loops</p></div></div>
    </div>
    <div class="row param-docs-row border-bottom align-items-center ">
      <div class="col-10 col-md-5 col-lg-4 col-xl-3 small-h ps-3 param-docs-row-id-col"><h3><code>--do_tfea</code></h3></div>
      <div class="col me-auto">
                  <input id="do_tfea" name="do_tfea" type="checkbox" class="hicarinput" >
              </div>
      <div class="col-auto d-flex flex-column align-items-end my-1"><div id="--do_tfea-body" class="param-docs-body small"><p>Do Transcription Factor Enrichment Analysis or not</p></div></div>
    </div>
    <div class="row param-docs-row border-bottom align-items-center ">
      <div class="col-10 col-md-5 col-lg-4 col-xl-3 small-h ps-3 param-docs-row-id-col"><h3><code>--create_virtual_4c</code></h3></div>
      <div class="col me-auto">
                  <input id="create_virtual_4c" name="create_virtual_4c" type="checkbox" class="hicarinput" >
              </div>
      <div class="col-auto d-flex flex-column align-items-end my-1"><div id="--create_virtual_4c-body" class="param-docs-body small"><p>create track files for virtual 4c or not</p></div></div>
    </div>
    <div class="row param-docs-row border-bottom align-items-center ">
      <div class="col-10 col-md-5 col-lg-4 col-xl-3 small-h ps-3 param-docs-row-id-col"><h3><code>--skip_circos</code></h3></div>
      <div class="col me-auto">
                  <input id="skip_circos" name="skip_circos" type="checkbox" class="hicarinput" >
              </div>
      <div class="col-auto d-flex flex-column align-items-end my-1"><div id="--skip_circos-body" class="param-docs-body small"><p>skip circos plot</p></div></div>
    </div>
  </from>
  
      <div class="col-12 param-docs-row-id-col">
          <h2 id="inputoutput-options">
              <i class="fas fa-terminal fa-fw"></i>
              Example of profile.config
          </h2>
      </div>
      <div class="alert border bg-light small">
        <pre>
// submit by slurm
process.executor = "slurm"
process.clusterOptions = "-J SomeJob -p scavenger"

params {
    // Limit resources
    max_cpus = 16
    max_memory = 240.GB
    max_time   = 240.h
}

process {
    withName: PAIRTOOLS_SORT {
        memory        = 120.GB
    }
    withName: MAPS_CALLPEAK {
        memory        = 120.GB
    }
}
        </pre>
  </div>
  </div>
    <script>
    $(document).ready(function(){
      $(".genomeoption").hide();
      $("#cmdform").validate({
        rules:{
      input:{pattern:"^\S+\.(csv|tsv|yaml)$",required:true},
      anchor_peaks:{pattern:"^\S+\.(narrowPeak|boradPeak)$"},
      outdir:{required:true},
      email:{pattern:"^([a-zA-Z0-9_\-\.]+)@([a-zA-Z0-9_\-\.]+)\.([a-zA-Z]{2,5})$"},
      fasta:{pattern:"^\S+\.fn?a(sta)?(\.gz)?$"},
      max_memory:{pattern:"^\d+(\.\d+)?\.?\s*(K|M|G|T)?B$"},
      max_time:{pattern:"^(\d+\.?\s*(s|m|h|d|day)\s*)+$"},
      email_on_fail:{pattern:"^([a-zA-Z0-9_\-\.]+)@([a-zA-Z0-9_\-\.]+)\.([a-zA-Z]{2,5})$"},
      max_multiqc_email_size:{pattern:"^\d+(\.\d+)?\.?\s*(K|M|G|T)?B$"}}
      });
      var defaultvalues={"input":"","method":'HiCAR',"anchor_peaks":"","outdir":"","email":"","multiqc_title":"","genome":"","fasta":"","bwa_index":"","gtf":"","gff":"","gene_bed":"","mappability":"","macs_gsize":"","ucscname":"","blacklist":"","publish_mappability":"","publish_genome":"","enzyme":'CviQI',"restriction_sites_cut_off":0.5,"cutadapt_5end":'^TAC',"cutadapt_action":'none',"cutadapt_error_tolerance":0.15,"shiftsize":-75,"smooth_window":150,"qval_thresh":0.01,"cool_bin":'5000_10000',"maps_digest_file":'None',"maps_cutoff_counts":12,"maps_cutoff_fold_change":2,"maps_cutoff_fdr":2,"maps_filter":'None',"maps_model":'pospoisson',"snow_type":'SOCK',"hicdcplus_cutoff_fdr":0.05,"peakachu_pretrained_url":'http://3dgenome.fsm.northwestern.edu/peakachu/HiCAR-models/HiCAR-peakachu-pretrained.',"call_high_peak":"","r1_pval_thresh":0.1,"res_compartments":100000,"res_tads":10000,"tad_tool":'hicexplorer',"apa_peak":'Anchor 1D peaks',"apa_tool":'hicexplorer',"apa_format":'png',"compartments_tool":'cooltools',"interactions_tool":'maps',"da_tool":'diffhic',"v4c_tool":'hicexplorer',"tfea_tool":'homer',"juicer_tools_jar":'https://github.com/aidenlab/JuicerTools/releases/download/v3.0.0/juicer_tools.3.0.0.jar',"juicer_norm_method":'SCALE',"peak_interactions_threshold":2,"v4c_max_events":25,"hic_tools_jar":'https://github.com/aidenlab/HiCTools/releases/download/v3.30.00/hic_tools.3.30.00.jar',"multiqc_methods_description":"","skip_cutadapt":"","resample_pairs":"","skip_fastqc":"","skip_peak_annotation":"","skip_diff_analysis":"","skip_multiqc":"","do_apa":"","skip_compartments":"","skip_tads":"","skip_interactions":"","do_tfea":"","create_virtual_4c":"","skip_circos":"","profile":"","config":""};
      getCmd = function(){
        var cmd = "nextflow run jianhong/hicar -r " + $("#version_select").val();
        $(".hicarinput").each(function(index, element){
          if($(this).val()!="" && $(this).val()!=defaultvalues[$(this).attr("id")]){
            const params0 = ["profile", "c"];
            const params01 = ["profile", "config"];
            if($.inArray($(this).attr("id"), params01)!== -1){
              for(let i=0; i<params01.length; i++){
                if($(this).attr("id")==params01[i]){
                  cmd += " -" + params0[i] + " " + $(this).val();
                }
              }
            }else{
              if($(this).attr("type")=="checkbox"){
                if($(this).is(":checked")){
                  cmd += " --" + $(this).attr("id");
                }
              }else{
                cmd += " --" + $(this).attr("id") + " " + $(this).val();
              }
            }
          }else{
            if($(this).attr("id")=="input"){
              cmd += " --input required";
            }
          }
        });


        return(cmd);
      }
      $("#code").text(getCmd());
      $(".hicarinput").change(function(){
        var cmd = getCmd();
        $("#code").text(cmd);
      });
     function readTopNLines(file, maxlines) {
        var CHUNK_SIZE = 50000; // 50kb, arbitrarily chosen.
        var decoder = new TextDecoder();
        var offset = 0;
        var linecount = 0;
        var linenumber = 0;
        var results = "";
        var fr = new FileReader();
        var counts =0;
        forEachLine = function(line, i){
                  if(i % 4 == 1){
                      return(new RegExp($("#cutadapt_5end").val()).test(line)?1:0);
                  }
                  return(0);
              };
        onComplete = function(counts){
            var rn = Math. floor(maxlines/4);
            counts = 100*counts/rn;
            var prefix = "";
            if(counts>50){
                prefix = "Congratulations! No issue detected.";
            }else{
                prefix = "Sorry, there may have some issues with your cutadapt parameter!.";
            }
            alert(prefix + "\nThere are " + counts + " % of top " + rn + " reads contains "+ $("#cutadapt_5end").val() + "!");
        }
        fr.onload = function() {
            // Use stream:true in case we cut the file
            // in the middle of a multi-byte character
            results += decoder.decode(fr.result, {stream: true});
            var lines = results.split("\n");
            results = lines.pop(); // In case the line did not end yet.
            linecount += lines.length;

            if (linecount > maxlines) {
                // Read too many lines? Truncate the results.
                lines.length -= linecount - maxlines;
                linecount = maxlines;
            }

            for (var i = 0; i < lines.length; ++i) {
                linenumber++;
                counts += forEachLine(lines[i] + "\n", linenumber-1);
            }
            offset += CHUNK_SIZE;
            seek();
        };
        fr.onerror = function() {
            alert("Wrong format of you input data!");
            onComplete(counts);
        };
        seek();

        function seek() {
            try{
                if (linecount === maxlines) {
                    // We found enough lines.
                    onComplete(counts); // Done.
                    return;
                }
                if (offset !== 0 && offset >= file.size) {
                    // We did not find all lines, but there are no more lines.
                    linenumber++;
                    counts += forEachLine(lines[i] + "\n", linenumber-1); // This is from lines.pop(), before.
                    onComplete(counts); // Done
                    return;
                }
                var slice = file.slice(offset, offset + CHUNK_SIZE);
                fr.readAsArrayBuffer(slice);
            }
            catch(e){
                alert("Wrong format of you input data!");
            }
        }
    }
    function checkAdapter(){
          if ($("#cutadapt_5end").val().length>5) {
              if($("#cutadapt_action").val()!="trim"){
                alert("The length of 5end adapter is greater than 5, changed the action into trim.");
              }
              $("#cutadapt_action").val("trim");
          }else{
              $("#cutadapt_action").val("none");
          }
          if (!window.File || !window.FileReader || !window.FileList || !window.Blob) {
              alert("The File APIs are not fully supported in this browser.");
              return;
          }
          var file = $("#fastq_r1").prop("files")[0];
          if(file){
              if($("#fastq_r1").val().split(".").pop().toLowerCase()=="gz"){
                  alert("The gz file is not support yet. Please try to load top 1000 lines by zcat f1_reads.gz > r1_test.fastq!");
                  return;
              }
              readTopNLines(file, 4000, function onComplete(){});
          }
      }
      $("#fastq_r1").change(checkAdapter);
      $("#cutadapt_5end").change(checkAdapter);
      function checkInput(){
          if (!window.File || !window.FileReader || !window.FileList || !window.Blob) {
              alert("The File APIs are not fully supported in this browser.");
              return;
          }
          var file = $("#input").prop("files")[0];
          if(file){
              var fr = new FileReader();
              fr.readAsText(file);
              fr.onload = function (evt) {
                  var data;
                  try{
                      data = $.csv.toObjects(evt.target.result);
                  }
                  catch(e){
                      alert("The sample must be a csv file.");
                      return;
                  }
                  const neededKeys = ["group", "replicate", "fastq_1", "fastq_2"];
                  var key_exist = neededKeys.every(key => Object.keys(data[0]).includes(key));
                  if(!key_exist){
                      alert("The sample information must contain the columns named as \"group\", \"replicate\", \"fastq_1\", and \"fastq_2\"");
                      return;
                  }
                  var samples = [];
                  data.forEach(function(it){
                      samples.push(it.group + it.replicate);
                  });
                  function unique(array){
                        return array.filter(function(el, index, arr) {
                            return index === arr.indexOf(el);
                        });
                  }
                  var len = samples.length;
                  var len2 = unique(samples).length;
                  if(len>len2){
                      if((!Object.keys(data[0]).includes("techniquereplicate")) && $("#version").val()=="dev"){
                          alert("The sample information must contain the columns named as \"techniquereplicate\"");
                          return;
                      }
                  }
              };
          }

      }
      $("#input").change(checkInput);
      $("#genome").change(function(){
          if($("#genome").val()=="custom"){
              $(".genomeoption").show();
          }else{
              $(".genomeoption").hide();
          }
      });
    });
    </script></div></div><div class="col-12 col-lg-3 ps-2"><div class="side-sub-subnav sticky-top"></div></div></div>